description: disable gecko/mono installer download
author: Michael Gilbert <mgilbert@debian.org>
author: Jens Reyer <jre.winesim@gmail.com>

--- a/dlls/appwiz.cpl/addons.c
+++ b/dlls/appwiz.cpl/addons.c
@@ -795,9 +795,8 @@ BOOL install_addon(addon_t addon_type)
      */
     if (install_from_registered_dir() == INSTALL_NEXT
         && install_from_default_dir() == INSTALL_NEXT
-        && install_from_cache() == INSTALL_NEXT
-        && (url = get_url()))
-        DialogBoxW(hInst, addon->dialog_template, 0, installer_proc);
+        && install_from_cache() == INSTALL_NEXT)
+        {};
 
     heap_free(url);
     url = NULL;
