--- wine.orig/tools/make_unicode	2020-06-24 18:20:08.000000000 +0300
+++ wine/tools/make_unicode	2020-06-24 18:27:12.000000000 +0300
@@ -172,7 +172,8 @@
     "Consonant_With_Stacker" => 0x001f,
     "Consonant_Prefixed" => 0x0020,
     "Syllable_Modifier" => 0x0021,
-    "Consonant_Killer" => 0x0022
+    "Consonant_Killer" => 0x0022,
+    "Consonant_Initial_Postfixed" => 0x0023,
 );
 
 my %matra_types =
@@ -1455,7 +1456,12 @@
     "Old_Hungarian"          => 134,
     "Osage"                  => 135,
     "SignWriting"            => 136,
-    "Tangut"                 => 137
+    "Tangut"                 => 137,
+    # Win10 RS4
+    "Masaram_Gondi"          => 138,
+    "Nushu"                  => 139,
+    "Soyombo"                => 140,
+    "Zanabazar_Square"       => 141,
 );
 
 ################################################################
