diff -ruN gtk+3.0-3.24.5.orig/modules/printbackends/cups/gtkprintbackendcups.c gtk+3.0-3.24.5/modules/printbackends/cups/gtkprintbackendcups.c
--- gtk+3.0-3.24.5.orig/modules/printbackends/cups/gtkprintbackendcups.c	2019-01-31 10:43:01.000000000 +0200
+++ gtk+3.0-3.24.5/modules/printbackends/cups/gtkprintbackendcups.c	2021-09-01 12:28:30.943330445 +0300
@@ -3754,6 +3754,7 @@
 cups_get_printer_list (GtkPrintBackend *backend)
 {
   GtkPrintBackendCups *cups_backend;
+  char *show_avahi_printers;
 
   cups_backend = GTK_PRINT_BACKEND_CUPS (backend);
 
@@ -3771,7 +3772,10 @@
         }
 
 #ifdef HAVE_CUPS_API_1_6
-      avahi_request_printer_list (cups_backend);
+      show_avahi_printers = g_getenv("GTK_SHOW_AVAHI_PRINTERS");
+      if (!(show_avahi_printers != NULL && strcmp(show_avahi_printers, "false") == 0)) {
+        avahi_request_printer_list (cups_backend);
+      }
 #endif
     }
 }
