description: don't append wine to the default path settings
author: Michael Gilbert <mgilbert@debian.org>

--- a/configure.ac
+++ b/configure.ac
@@ -672,7 +672,7 @@ AC_CHECK_HEADERS([libprocstat.h],,,
 
 dnl **** Check for working dll ****
 
-AC_SUBST(dlldir,"\${libdir}/wine")
+AC_SUBST(dlldir,"\${libdir}")
 AC_SUBST(DLLFLAGS,"-D_REENTRANT")
 AC_SUBST(LDRPATH_INSTALL,"")
 AC_SUBST(LDRPATH_LOCAL,"")
--- a/Makefile.in
+++ b/Makefile.in
@@ -27,8 +27,8 @@ libdir          = @libdir@
 datarootdir     = @datarootdir@
 datadir         = @datadir@
 mandir          = @mandir@
-fontdir         = ${datadir}/wine/fonts
-includedir      = @includedir@/wine
+fontdir         = /usr/share/wine/fonts
+includedir      = @includedir@
 dlldir          = @dlldir@
 fakedlldir      = ${dlldir}/fakedlls
 top_srcdir      = @top_srcdir@
