Description: Use system-wide quazip headers.
Author: Mike Gabriel <mike.gabriel@das-netzwerkteam.de>

--- a/src/adaptors/UBExportDocument.cpp
+++ b/src/adaptors/UBExportDocument.cpp
@@ -39,10 +39,8 @@
 
 #include "globals/UBGlobals.h"
 
-THIRD_PARTY_WARNINGS_DISABLE
-#include "quazip.h"
-#include "quazipfile.h"
-THIRD_PARTY_WARNINGS_ENABLE
+#include <quazip5/quazip.h>
+#include <quazip5/quazipfile.h>
 
 #include "core/memcheck.h"
 
--- a/src/adaptors/UBExportDocumentSetAdaptor.cpp
+++ b/src/adaptors/UBExportDocumentSetAdaptor.cpp
@@ -40,10 +40,8 @@
 #include "core/UBPersistenceManager.h"
 #include "core/UBForeignObjectsHandler.h"
 
-THIRD_PARTY_WARNINGS_DISABLE
-#include "quazip.h"
-#include "quazipfile.h"
-THIRD_PARTY_WARNINGS_ENABLE
+#include <quazip5/quazip.h>
+#include <quazip5/quazipfile.h>
 
 #include "core/memcheck.h"
 
--- a/src/adaptors/UBExportDocumentSetAdaptor.h
+++ b/src/adaptors/UBExportDocumentSetAdaptor.h
@@ -33,10 +33,8 @@
 #include "frameworks/UBFileSystemUtils.h"
 #include "globals/UBGlobals.h"
 
-THIRD_PARTY_WARNINGS_DISABLE
-#include "quazip.h"
-#include "quazipfile.h"
-THIRD_PARTY_WARNINGS_ENABLE
+#include <quazip5/quazip.h>
+#include <quazip5/quazipfile.h>
 
 class UBDocumentProxy;
 class UBDocumentTreeModel;
--- a/src/adaptors/UBExportWeb.cpp
+++ b/src/adaptors/UBExportWeb.cpp
@@ -39,10 +39,8 @@
 
 #include "globals/UBGlobals.h"
 
-THIRD_PARTY_WARNINGS_DISABLE
-#include "quazip.h"
-#include "quazipfile.h"
-THIRD_PARTY_WARNINGS_ENABLE
+#include <quazip5/quazip.h>
+#include <quazip5/quazipfile.h>
 
 #include "core/memcheck.h"
 
--- a/src/adaptors/UBImportCFF.cpp
+++ b/src/adaptors/UBImportCFF.cpp
@@ -43,11 +43,9 @@
 
 #include "globals/UBGlobals.h"
 
-THIRD_PARTY_WARNINGS_DISABLE
-#include "quazip.h"
-#include "quazipfile.h"
-#include "quazipfileinfo.h"
-THIRD_PARTY_WARNINGS_ENABLE
+#include <quazip5/quazip.h>
+#include <quazip5/quazipfile.h>
+#include <quazip5/quazipfileinfo.h>
 
 #include "core/memcheck.h"
 
--- a/src/adaptors/UBImportDocument.cpp
+++ b/src/adaptors/UBImportDocument.cpp
@@ -38,11 +38,9 @@
 
 #include "globals/UBGlobals.h"
 
-THIRD_PARTY_WARNINGS_DISABLE
-#include "quazip.h"
-#include "quazipfile.h"
-#include "quazipfileinfo.h"
-THIRD_PARTY_WARNINGS_ENABLE
+#include <quazip5/quazip.h>
+#include <quazip5/quazipfile.h>
+#include <quazip5/quazipfileinfo.h>
 
 #include "core/memcheck.h"
 
--- a/src/adaptors/UBImportDocumentSetAdaptor.cpp
+++ b/src/adaptors/UBImportDocumentSetAdaptor.cpp
@@ -36,11 +36,9 @@
 
 #include "globals/UBGlobals.h"
 
-THIRD_PARTY_WARNINGS_DISABLE
-#include "quazip.h"
-#include "quazipfile.h"
-#include "quazipfileinfo.h"
-THIRD_PARTY_WARNINGS_ENABLE
+#include <quazip5/quazip.h>
+#include <quazip5/quazipfile.h>
+#include <quazip5/quazipfileinfo.h>
 
 #include "core/memcheck.h"
 
--- a/src/frameworks/UBFileSystemUtils.cpp
+++ b/src/frameworks/UBFileSystemUtils.cpp
@@ -35,10 +35,8 @@
 
 #include "globals/UBGlobals.h"
 
-THIRD_PARTY_WARNINGS_DISABLE
-#include "quazipfile.h"
+#include <quazip5/quazipfile.h>
 #include <openssl/md5.h>
-THIRD_PARTY_WARNINGS_ENABLE
 
 #include "core/memcheck.h"
 
--- a/plugins/cffadaptor/src/UBCFFAdaptor.cpp
+++ b/plugins/cffadaptor/src/UBCFFAdaptor.cpp
@@ -33,11 +33,9 @@
 #include "UBGlobals.h"
 #include "UBCFFConstants.h"
 
-//THIRD_PARTY_WARNINGS_DISABLE
-#include "quazip.h"
-#include "quazipfile.h"
-#include "quazipfileinfo.h"
-//THIRD_PARTY_WARNINGS_ENABLE
+#include <quazip5/quazip.h>
+#include <quazip5/quazipfile.h>
+#include <quazip5/quazipfileinfo.h>
 
 UBCFFAdaptor::UBCFFAdaptor()
 {}
