Description: Correctly show stencil color. Always.
Author: Claudio Laurita <claudio.laurita@integrazionetotale.it>

--- a/src/core/UBApplication.cpp
+++ b/src/core/UBApplication.cpp
@@ -301,8 +301,6 @@
     webController = new UBWebController(mainWindow);
     documentController = new UBDocumentController(mainWindow);
 
-    UBDrawingController::drawingController()->setStylusTool((int)UBStylusTool::Pen);
-
     applicationController = new UBApplicationController(boardController->controlView(),
                                                         boardController->displayView(),
                                                         mainWindow,
@@ -358,6 +356,8 @@
 
     emit UBDrawingController::drawingController()->colorPaletteChanged();
 
+    UBDrawingController::drawingController()->setStylusTool((int)UBStylusTool::Pen);
+
     onScreenCountChanged(1);
     connect(desktop(), SIGNAL(screenCountChanged(int)), this, SLOT(onScreenCountChanged(int)));
     return QApplication::exec();
