Description: Correct Qt Library name for QtWebKit.
Author: Mike Gabriel <mike.gabriel@das-netzwerkteam.de>

--- a/resources/forms/trapFlash.ui
+++ b/resources/forms/trapFlash.ui
@@ -113,7 +113,7 @@
   <customwidget>
    <class>QWebView</class>
    <extends>QWidget</extends>
-   <header>QtWebKit/QWebView</header>
+   <header>QtWebKitWidgets/QWebView</header>
   </customwidget>
  </customwidgets>
  <resources/>
