Description: Various typo fixes, spotted by Debian's lintian tool.
Author: Mike Gabriel <mike.gabriel@das-netzwerkteam.de>

--- a/resources/forms/preferences.ui
+++ b/resources/forms/preferences.ui
@@ -1157,7 +1157,7 @@
  * The implementation was written so as to conform with Netscapes SSL.
  *
  * This library is free for commercial and non-commercial use as long as
- * the following conditions are aheared to.  The following conditions
+ * the following conditions are adhered to.  The following conditions
  * apply to all code found in this distribution, be it the RC4, RSA,
  * lhash, DES, etc., code; not just the SSL code.  The SSL documentation
  * included with this distribution is covered by the same copyright terms
@@ -1182,7 +1182,7 @@
  *    must display the following acknowledgement:
  *    &quot;This product includes cryptographic software written by
  *     Eric Young (eay@cryptsoft.com)&quot;
- *    The word 'cryptographic' can be left out if the rouines from the library
+ *    The word 'cryptographic' can be left out if the routines from the library
  *    being used are not cryptographic related :-).
  * 4. If you include any Windows specific code (or a derivative thereof) from
  *    the apps directory (application code) you must include an acknowledgement:
@@ -1203,7 +1203,7 @@
  * The licence and distribution terms for any publically available version or
  * derivative of this code cannot be changed.  i.e. this code cannot simply be
  * copied and put under another distribution licence
- * [including the GNU Public Licence.]
+ * [including the GNU General Public License.]
  */</string>
              </property>
              <property name="textInteractionFlags">
--- a/plugins/cffadaptor/src/UBCFFAdaptor.cpp
+++ b/plugins/cffadaptor/src/UBCFFAdaptor.cpp
@@ -42,15 +42,15 @@
 
 bool UBCFFAdaptor::convertUBZToIWB(const QString &from, const QString &to)
 {
-    qDebug() << "starting converion from" << from << "to" << to;
+    qDebug() << "starting conversion from" << from << "to" << to;
 
     QString source = QString();
     if (QFileInfo(from).isDir() && QFile::exists(from)) {
-        qDebug() << "File specified is dir, continuing convertion";
+        qDebug() << "File specified is dir, continuing conversion";
         source = from;
     } else {
         source = uncompressZip(from);
-        if (!source.isNull()) qDebug() << "File specified is zip file. Uncompressed to tmp dir, continuing convertion";
+        if (!source.isNull()) qDebug() << "File specified is zip file. Uncompressed to tmp dir, continuing conversion";
     }
     if (source.isNull()) {
         qDebug() << "File specified is not a dir or a zip file, stopping covretion";
--- a/resources/i18n/OpenBoard_ar.ts
+++ b/resources/i18n/OpenBoard_ar.ts
@@ -1272,7 +1272,7 @@
         <translation type="unfinished"></translation>
     </message>
     <message>
-        <source>The name %1 is allready used.
+        <source>The name %1 is already used.
 Keeping this name will replace the document.
 Providing a new name will create a new document.</source>
         <translation type="unfinished"></translation>
@@ -1561,7 +1561,7 @@
         <translation>الاضافة الى المكتبة</translation>
     </message>
     <message>
-        <source>Object informations</source>
+        <source>Object information</source>
         <translation>معلومات حول الأشياء</translation>
     </message>
 </context>
@@ -2189,7 +2189,7 @@
         <translation>اختيار مجلد النسخ الاحتياطي</translation>
     </message>
     <message>
-        <source>An error occured during the update. The files have not been affected.</source>
+        <source>An error occurred during the update. The files have not been affected.</source>
         <translation>حدث خطأ أثناء التحديث. لم يقع التأثير على الملفات.</translation>
     </message>
     <message>
--- a/resources/i18n/OpenBoard_bg.ts
+++ b/resources/i18n/OpenBoard_bg.ts
@@ -1257,7 +1257,7 @@
         <translation type="unfinished"></translation>
     </message>
     <message>
-        <source>The name %1 is allready used.
+        <source>The name %1 is already used.
 Keeping this name will replace the document.
 Providing a new name will create a new document.</source>
         <translation type="unfinished"></translation>
@@ -1534,7 +1534,7 @@
         <translation>Добави към библиотека</translation>
     </message>
     <message>
-        <source>Object informations</source>
+        <source>Object information</source>
         <translation>Информация за обекта</translation>
     </message>
 </context>
@@ -2173,7 +2173,7 @@
         <translation>Обновяването на файловете е успешно! Моля рестартирайте приложението, за да имате достъп до обновените документи.</translation>
     </message>
     <message>
-        <source>An error occured during the update. The files have not been affected.</source>
+        <source>An error occurred during the update. The files have not been affected.</source>
         <translation>Възникна грешка по време на обновяването. Файловете не са засегнати.</translation>
     </message>
     <message>
--- a/resources/i18n/OpenBoard_ca.ts
+++ b/resources/i18n/OpenBoard_ca.ts
@@ -1263,7 +1263,7 @@
         <translation type="unfinished"></translation>
     </message>
     <message>
-        <source>The name %1 is allready used.
+        <source>The name %1 is already used.
 Keeping this name will replace the document.
 Providing a new name will create a new document.</source>
         <translation type="unfinished"></translation>
@@ -1540,7 +1540,7 @@
         <translation>Afegeix a la biblioteca</translation>
     </message>
     <message>
-        <source>Object informations</source>
+        <source>Object information</source>
         <translation>Informacions de l&apos;objecte</translation>
     </message>
 </context>
@@ -2173,7 +2173,7 @@
 Reinicieu l&apos;aplicació per accedir als documents actualitzats.</translation>
     </message>
     <message>
-        <source>An error occured during the update. The files have not been affected.</source>
+        <source>An error occurred during the update. The files have not been affected.</source>
         <translation>Ha ocorregut un error durant l&apos;actualització. Els fitxers no s&apos;han vist afectats.</translation>
     </message>
     <message>
--- a/resources/i18n/OpenBoard_cs.ts
+++ b/resources/i18n/OpenBoard_cs.ts
@@ -1259,7 +1259,7 @@
         <translation type="unfinished"></translation>
     </message>
     <message>
-        <source>The name %1 is allready used.
+        <source>The name %1 is already used.
 Keeping this name will replace the document.
 Providing a new name will create a new document.</source>
         <translation type="unfinished"></translation>
@@ -1539,7 +1539,7 @@
         <translation>Přidat do knihovny</translation>
     </message>
     <message>
-        <source>Object informations</source>
+        <source>Object information</source>
         <translation>Informace o objektu</translation>
     </message>
 </context>
@@ -2179,7 +2179,7 @@
         <translation>Aktualizovat</translation>
     </message>
     <message>
-        <source>An error occured during the update. The files have not been affected.</source>
+        <source>An error occurred during the update. The files have not been affected.</source>
         <translation>Během aktualizace se vyskytla chyba. Soubory to neovlivnilo.</translation>
     </message>
     <message>
--- a/resources/i18n/OpenBoard_da.ts
+++ b/resources/i18n/OpenBoard_da.ts
@@ -1263,7 +1263,7 @@
         <translation type="unfinished"></translation>
     </message>
     <message>
-        <source>The name %1 is allready used.
+        <source>The name %1 is already used.
 Keeping this name will replace the document.
 Providing a new name will create a new document.</source>
         <translation type="unfinished"></translation>
@@ -1540,7 +1540,7 @@
         <translation>Tilføj til bibliotek</translation>
     </message>
     <message>
-        <source>Object informations</source>
+        <source>Object information</source>
         <translation>Objektinformationer</translation>
     </message>
 </context>
@@ -2180,7 +2180,7 @@
 Genstart venligst applikationen for at få adgang til de opdaterede dokumenter.</translation>
     </message>
     <message>
-        <source>An error occured during the update. The files have not been affected.</source>
+        <source>An error occurred during the update. The files have not been affected.</source>
         <translation>Der opstod en fejl under opdateringen. Filerne er ikke påvirket af fejlen.</translation>
     </message>
     <message>
--- a/resources/i18n/OpenBoard_de.ts
+++ b/resources/i18n/OpenBoard_de.ts
@@ -1275,7 +1275,7 @@
         <translation>Abbrechen</translation>
     </message>
     <message>
-        <source>The name %1 is allready used.
+        <source>The name %1 is already used.
 Keeping this name will replace the document.
 Providing a new name will create a new document.</source>
         <translation>Der Name %1 wird bereits benutzt.
@@ -1578,7 +1578,7 @@
         <translation>Zur Bibliothek hinzufügen</translation>
     </message>
     <message>
-        <source>Object informations</source>
+        <source>Object information</source>
         <translation>Objektinformationen</translation>
     </message>
 </context>
@@ -2239,7 +2239,7 @@
 Bitte starten Sie die Anwendung erneut, um auf die aktualisierten Dokumente zugreifen zu können.</translation>
     </message>
     <message>
-        <source>An error occured during the update. The files have not been affected.</source>
+        <source>An error occurred during the update. The files have not been affected.</source>
         <translation>Während des Updates ist ein Fehler aufgetreten. Die Dateien wurden nicht beeinträchtigt.</translation>
     </message>
     <message>
--- a/resources/i18n/OpenBoard_el.ts
+++ b/resources/i18n/OpenBoard_el.ts
@@ -1256,7 +1256,7 @@
         <translation type="unfinished"></translation>
     </message>
     <message>
-        <source>The name %1 is allready used.
+        <source>The name %1 is already used.
 Keeping this name will replace the document.
 Providing a new name will create a new document.</source>
         <translation type="unfinished"></translation>
@@ -1533,7 +1533,7 @@
         <translation>Προσθήκη στη βιβλιοθήκη</translation>
     </message>
     <message>
-        <source>Object informations</source>
+        <source>Object information</source>
         <translation>Πληροφορίες αντικειμένου</translation>
     </message>
 </context>
@@ -2173,7 +2173,7 @@
 Παρακαλώ επανεκκινήστε την εφαρμογή για να έχετε πρόσβαση στα ενημερωμένα αρχεία.</translation>
     </message>
     <message>
-        <source>An error occured during the update. The files have not been affected.</source>
+        <source>An error occurred during the update. The files have not been affected.</source>
         <translation>Συνέβη ένα σφάλμα κατά την ενημέρωση. Τα αρχεία δεν έχουν επηρεαστεί.</translation>
     </message>
     <message>
--- a/resources/i18n/OpenBoard_en.ts
+++ b/resources/i18n/OpenBoard_en.ts
@@ -1204,7 +1204,7 @@
         <translation type="unfinished"></translation>
     </message>
     <message>
-        <source>The name %1 is allready used.
+        <source>The name %1 is already used.
 Keeping this name will replace the document.
 Providing a new name will create a new document.</source>
         <translation type="unfinished"></translation>
@@ -1433,7 +1433,7 @@
         <translation type="unfinished"></translation>
     </message>
     <message>
-        <source>Object informations</source>
+        <source>Object information</source>
         <translation type="unfinished"></translation>
     </message>
 </context>
@@ -2040,7 +2040,7 @@
         <translation type="unfinished"></translation>
     </message>
     <message>
-        <source>An error occured during the update. The files have not been affected.</source>
+        <source>An error occurred during the update. The files have not been affected.</source>
         <translation type="unfinished"></translation>
     </message>
     <message>
--- a/resources/i18n/OpenBoard_en_UK.ts
+++ b/resources/i18n/OpenBoard_en_UK.ts
@@ -1204,7 +1204,7 @@
         <translation type="unfinished"></translation>
     </message>
     <message>
-        <source>The name %1 is allready used.
+        <source>The name %1 is already used.
 Keeping this name will replace the document.
 Providing a new name will create a new document.</source>
         <translation type="unfinished"></translation>
@@ -1433,7 +1433,7 @@
         <translation type="unfinished"></translation>
     </message>
     <message>
-        <source>Object informations</source>
+        <source>Object information</source>
         <translation type="unfinished"></translation>
     </message>
 </context>
@@ -2040,7 +2040,7 @@
         <translation type="unfinished"></translation>
     </message>
     <message>
-        <source>An error occured during the update. The files have not been affected.</source>
+        <source>An error occurred during the update. The files have not been affected.</source>
         <translation type="unfinished"></translation>
     </message>
     <message>
--- a/resources/i18n/OpenBoard_es.ts
+++ b/resources/i18n/OpenBoard_es.ts
@@ -1268,7 +1268,7 @@
         <translation>Cancelar</translation>
     </message>
     <message>
-        <source>The name %1 is allready used.
+        <source>The name %1 is already used.
 Keeping this name will replace the document.
 Providing a new name will create a new document.</source>
         <translation>El nombre %1 ya está en uso.
@@ -1547,7 +1547,7 @@
         <translation>Añadir a biblioteca</translation>
     </message>
     <message>
-        <source>Object informations</source>
+        <source>Object information</source>
         <translation>Información del objeto</translation>
     </message>
 </context>
@@ -2193,7 +2193,7 @@
 Por favor, reinicie la aplicación para acceder a los documentos actualizados.</translation>
     </message>
     <message>
-        <source>An error occured during the update. The files have not been affected.</source>
+        <source>An error occurred during the update. The files have not been affected.</source>
         <translation>Ha ocurrido un error durante la actualización. Los archivos no han sido afectados.</translation>
     </message>
     <message>
--- a/resources/i18n/OpenBoard_eu.ts
+++ b/resources/i18n/OpenBoard_eu.ts
@@ -1333,7 +1333,7 @@
         <translation type="vanished">Gehitu liburutegira</translation>
     </message>
     <message>
-        <source>Object informations</source>
+        <source>Object information</source>
         <translation type="vanished">Objektuaren informazioa</translation>
     </message>
 </context>
@@ -1929,7 +1929,7 @@
 Mesedez berrabiarazi aplikazioa eguneratutako dokumentuak irekitzeko.</translation>
     </message>
     <message>
-        <source>An error occured during the update. The files have not been affected.</source>
+        <source>An error occurred during the update. The files have not been affected.</source>
         <translation type="vanished">Errorea gertatu da eguneraketan zehar. Fitxategiak ez dira kaltetuak izan.</translation>
     </message>
     <message>
--- a/resources/i18n/OpenBoard_fr.ts
+++ b/resources/i18n/OpenBoard_fr.ts
@@ -1285,7 +1285,7 @@
         <translation>Annuler</translation>
     </message>
     <message>
-        <source>The name %1 is allready used.
+        <source>The name %1 is already used.
 Keeping this name will replace the document.
 Providing a new name will create a new document.</source>
         <translation>Le nom %1 est déjà utilisé.
@@ -1591,7 +1591,7 @@
         <translation>Ajout bibliothèque</translation>
     </message>
     <message>
-        <source>Object informations</source>
+        <source>Object information</source>
         <translation>Informations</translation>
     </message>
 </context>
@@ -2251,7 +2251,7 @@
 Veuillez redémarrer l&apos;application pour accéder aux documents mis à jour.</translation>
     </message>
     <message>
-        <source>An error occured during the update. The files have not been affected.</source>
+        <source>An error occurred during the update. The files have not been affected.</source>
         <translation>Une erreur s&apos;est produite durant la mise à jour. Les fichiers n&apos;ont pas été affectés.</translation>
     </message>
     <message>
--- a/resources/i18n/OpenBoard_fr_CH.ts
+++ b/resources/i18n/OpenBoard_fr_CH.ts
@@ -1285,7 +1285,7 @@
         <translation>Annuler</translation>
     </message>
     <message>
-        <source>The name %1 is allready used.
+        <source>The name %1 is already used.
 Keeping this name will replace the document.
 Providing a new name will create a new document.</source>
         <translation>Le nom %1 est déjà utilisé.
@@ -1591,7 +1591,7 @@
         <translation>Ajout bibliothèque</translation>
     </message>
     <message>
-        <source>Object informations</source>
+        <source>Object information</source>
         <translation>Informations</translation>
     </message>
 </context>
@@ -2251,7 +2251,7 @@
 Veuillez redémarrer l&apos;application pour accéder aux documents mis à jour.</translation>
     </message>
     <message>
-        <source>An error occured during the update. The files have not been affected.</source>
+        <source>An error occurred during the update. The files have not been affected.</source>
         <translation>Une erreur s&apos;est produite durant la mise à jour. Les fichiers n&apos;ont pas été affectés.</translation>
     </message>
     <message>
--- a/resources/i18n/OpenBoard_gl.ts
+++ b/resources/i18n/OpenBoard_gl.ts
@@ -1268,7 +1268,7 @@
         <translation>Cancelar</translation>
     </message>
     <message>
-        <source>The name %1 is allready used.
+        <source>The name %1 is already used.
 Keeping this name will replace the document.
 Providing a new name will create a new document.</source>
         <translation>O nome %1 xa está en uso.
@@ -1547,7 +1547,7 @@
         <translation>Engadir á biblioteca</translation>
     </message>
     <message>
-        <source>Object informations</source>
+        <source>Object information</source>
         <translation>Información do obxecto</translation>
     </message>
 </context>
@@ -2193,7 +2193,7 @@
 Por favor, reinicie a aplicación para acceder aos documentos actualizados.</translation>
     </message>
     <message>
-        <source>An error occured during the update. The files have not been affected.</source>
+        <source>An error occurred during the update. The files have not been affected.</source>
         <translation>Ocurriu un erro durante a actualización. Os archivos no foron afectados.</translation>
     </message>
     <message>
--- a/resources/i18n/OpenBoard_hu.ts
+++ b/resources/i18n/OpenBoard_hu.ts
@@ -1262,7 +1262,7 @@
         <translation type="unfinished">Mégse</translation>
     </message>
     <message>
-        <source>The name %1 is allready used.
+        <source>The name %1 is already used.
 Keeping this name will replace the document.
 Providing a new name will create a new document.</source>
         <translation type="unfinished"></translation>
@@ -1508,7 +1508,7 @@
         <translation>Hozzáadás könyvtárhoz</translation>
     </message>
     <message>
-        <source>Object informations</source>
+        <source>Object information</source>
         <translation>Objektum információk</translation>
     </message>
 </context>
@@ -2159,7 +2159,7 @@
         <translation>A fájlok frissítése sikeres. Kérem, hogy indítsa újra az alkalmazást, hogy hozzáférhessen a frissített dokumentumokhoz.</translation>
     </message>
     <message>
-        <source>An error occured during the update. The files have not been affected.</source>
+        <source>An error occurred during the update. The files have not been affected.</source>
         <translation>A frissítés közben hiba lépett fel. A fájlok nem sérültek.</translation>
     </message>
     <message>
--- a/resources/i18n/OpenBoard_it.ts
+++ b/resources/i18n/OpenBoard_it.ts
@@ -1275,7 +1275,7 @@
         <translation>Annulla</translation>
     </message>
     <message>
-        <source>The name %1 is allready used.
+        <source>The name %1 is already used.
 Keeping this name will replace the document.
 Providing a new name will create a new document.</source>
         <translation>Il nome %1 è già usato.
@@ -1578,7 +1578,7 @@
         <translation>Aggiungi alla Biblioteca</translation>
     </message>
     <message>
-        <source>Object informations</source>
+        <source>Object information</source>
         <translation>Informazioni sull&apos;oggetto</translation>
     </message>
 </context>
@@ -2234,7 +2234,7 @@
 Si prega di riavviare l&apos;applicazione per accedere ai documenti aggiornati.</translation>
     </message>
     <message>
-        <source>An error occured during the update. The files have not been affected.</source>
+        <source>An error occurred during the update. The files have not been affected.</source>
         <translation>Si è verificato un errore durante l&apos;aggiornamento. I file non sono stati coinvolti.</translation>
     </message>
     <message>
--- a/resources/i18n/OpenBoard_iw.ts
+++ b/resources/i18n/OpenBoard_iw.ts
@@ -1257,7 +1257,7 @@
         <translation type="unfinished"></translation>
     </message>
     <message>
-        <source>The name %1 is allready used.
+        <source>The name %1 is already used.
 Keeping this name will replace the document.
 Providing a new name will create a new document.</source>
         <translation type="unfinished"></translation>
@@ -1534,7 +1534,7 @@
         <translation>הוסף לספרייה</translation>
     </message>
     <message>
-        <source>Object informations</source>
+        <source>Object information</source>
         <translation>מידע על האובייקט</translation>
     </message>
 </context>
@@ -2169,7 +2169,7 @@
 יש לאתחל את היישום כדי לגשת למסמכים המעודכנים.</translation>
     </message>
     <message>
-        <source>An error occured during the update. The files have not been affected.</source>
+        <source>An error occurred during the update. The files have not been affected.</source>
         <translation>אירעה שגיאה במהלך העדכון. הקבצים לא נפגעו.</translation>
     </message>
     <message>
--- a/resources/i18n/OpenBoard_ja.ts
+++ b/resources/i18n/OpenBoard_ja.ts
@@ -1250,7 +1250,7 @@
         <translation type="unfinished"></translation>
     </message>
     <message>
-        <source>The name %1 is allready used.
+        <source>The name %1 is already used.
 Keeping this name will replace the document.
 Providing a new name will create a new document.</source>
         <translation type="unfinished"></translation>
@@ -1524,7 +1524,7 @@
         <translation>ライブラリに追加</translation>
     </message>
     <message>
-        <source>Object informations</source>
+        <source>Object information</source>
         <translation>オブジェクトインフォーメーション</translation>
     </message>
 </context>
@@ -2165,7 +2165,7 @@
 アプリケーションを再起動して更新ファイルにアクセスしてください。</translation>
     </message>
     <message>
-        <source>An error occured during the update. The files have not been affected.</source>
+        <source>An error occurred during the update. The files have not been affected.</source>
         <translation>更新中にエラーが発生しました。ファイルに影響はありません。 </translation>
     </message>
     <message>
--- a/resources/i18n/OpenBoard_ko.ts
+++ b/resources/i18n/OpenBoard_ko.ts
@@ -1261,7 +1261,7 @@
         <translation type="unfinished"></translation>
     </message>
     <message>
-        <source>The name %1 is allready used.
+        <source>The name %1 is already used.
 Keeping this name will replace the document.
 Providing a new name will create a new document.</source>
         <translation type="unfinished"></translation>
@@ -1544,7 +1544,7 @@
         <translation type="unfinished">라이브러리에 추가</translation>
     </message>
     <message>
-        <source>Object informations</source>
+        <source>Object information</source>
         <translation type="unfinished">객체 정보</translation>
     </message>
 </context>
@@ -2184,7 +2184,7 @@
 업데이트된 문서에 액세스하려면 응용 프로그램을 재시작하십시오.</translation>
     </message>
     <message>
-        <source>An error occured during the update. The files have not been affected.</source>
+        <source>An error occurred during the update. The files have not been affected.</source>
         <translation>업데이트 도중 오류가 발생했습니다. 파일은 영향 받지 않았습니다.</translation>
     </message>
     <message>
--- a/resources/i18n/OpenBoard_mg.ts
+++ b/resources/i18n/OpenBoard_mg.ts
@@ -1263,7 +1263,7 @@
         <translation type="unfinished"></translation>
     </message>
     <message>
-        <source>The name %1 is allready used.
+        <source>The name %1 is already used.
 Keeping this name will replace the document.
 Providing a new name will create a new document.</source>
         <translation type="unfinished"></translation>
@@ -1540,7 +1540,7 @@
         <translation>Ampiana amin&apos;ny tahirim-boky</translation>
     </message>
     <message>
-        <source>Object informations</source>
+        <source>Object information</source>
         <translation>Mombamomban&apos;ny zavatra</translation>
     </message>
 </context>
@@ -2180,7 +2180,7 @@
 Avereno alefa ny rindr&apos;asa ahafahana mampiasa ny rakitra novoazana azafady.</translation>
     </message>
     <message>
-        <source>An error occured during the update. The files have not been affected.</source>
+        <source>An error occurred during the update. The files have not been affected.</source>
         <translation>Nisy tsy nety tamin&apos;ny fanavaozana. Tsy niova ny rakitra.</translation>
     </message>
     <message>
--- a/resources/i18n/OpenBoard_nb.ts
+++ b/resources/i18n/OpenBoard_nb.ts
@@ -1257,7 +1257,7 @@
         <translation type="unfinished"></translation>
     </message>
     <message>
-        <source>The name %1 is allready used.
+        <source>The name %1 is already used.
 Keeping this name will replace the document.
 Providing a new name will create a new document.</source>
         <translation type="unfinished"></translation>
@@ -1534,7 +1534,7 @@
         <translation>Legg til bibliotek</translation>
     </message>
     <message>
-        <source>Object informations</source>
+        <source>Object information</source>
         <translation>Objektopplysninger</translation>
     </message>
 </context>
@@ -2168,7 +2168,7 @@
         <translation>Filene ble oppdatert! Vennligst start applikasjonen på nytt for å få tilgang til de oppdaterte dokumentene.</translation>
     </message>
     <message>
-        <source>An error occured during the update. The files have not been affected.</source>
+        <source>An error occurred during the update. The files have not been affected.</source>
         <translation>En feil oppsto under oppdateringen. Filene har ikke blitt berørt av dette.</translation>
     </message>
     <message>
--- a/resources/i18n/OpenBoard_nl.ts
+++ b/resources/i18n/OpenBoard_nl.ts
@@ -1259,7 +1259,7 @@
         <translation type="unfinished"></translation>
     </message>
     <message>
-        <source>The name %1 is allready used.
+        <source>The name %1 is already used.
 Keeping this name will replace the document.
 Providing a new name will create a new document.</source>
         <translation type="unfinished"></translation>
@@ -1536,7 +1536,7 @@
         <translation>Aan bibliotheek toevoegen</translation>
     </message>
     <message>
-        <source>Object informations</source>
+        <source>Object information</source>
         <translation>Object informatie</translation>
     </message>
 </context>
@@ -2169,7 +2169,7 @@
         <translation>Mappen update gelukt! De applicatie opnieuw starten voor toegang tot de geüpdate documenten.</translation>
     </message>
     <message>
-        <source>An error occured during the update. The files have not been affected.</source>
+        <source>An error occurred during the update. The files have not been affected.</source>
         <translation>Er is een fout opgetreden tijden de update.De mappen zijnniet beschadigd.</translation>
     </message>
     <message>
--- a/resources/i18n/OpenBoard_pl.ts
+++ b/resources/i18n/OpenBoard_pl.ts
@@ -1269,7 +1269,7 @@
         <translation type="unfinished"></translation>
     </message>
     <message>
-        <source>The name %1 is allready used.
+        <source>The name %1 is already used.
 Keeping this name will replace the document.
 Providing a new name will create a new document.</source>
         <translation type="unfinished"></translation>
@@ -1550,7 +1550,7 @@
         <translation>Dodaj do biblioteki</translation>
     </message>
     <message>
-        <source>Object informations</source>
+        <source>Object information</source>
         <translation>Informacje o obiekcie</translation>
     </message>
 </context>
@@ -2190,7 +2190,7 @@
 Aby uzyskać dostęp do zaktualizowanych dokumentów, należy ponownie uruchomić aplikację.</translation>
     </message>
     <message>
-        <source>An error occured during the update. The files have not been affected.</source>
+        <source>An error occurred during the update. The files have not been affected.</source>
         <translation>W trakcie aktualizacji wystąpił błąd. Pliki nie zostały uszkodzone.</translation>
     </message>
     <message>
--- a/resources/i18n/OpenBoard_pt.ts
+++ b/resources/i18n/OpenBoard_pt.ts
@@ -1266,7 +1266,7 @@
         <translation type="unfinished"></translation>
     </message>
     <message>
-        <source>The name %1 is allready used.
+        <source>The name %1 is already used.
 Keeping this name will replace the document.
 Providing a new name will create a new document.</source>
         <translation type="unfinished"></translation>
@@ -1567,7 +1567,7 @@
         <translation>Adicionar à biblioteca</translation>
     </message>
     <message>
-        <source>Object informations</source>
+        <source>Object information</source>
         <translation>Informações do Objecto</translation>
     </message>
 </context>
@@ -2209,7 +2209,7 @@
 Por favor, reinicie o aplicativo para aceder aos documentos atualizados.</translation>
     </message>
     <message>
-        <source>An error occured during the update. The files have not been affected.</source>
+        <source>An error occurred during the update. The files have not been affected.</source>
         <translation>Ocorreu um erro durante a atualização. Os ficheiros não foram afetados.</translation>
     </message>
     <message>
--- a/resources/i18n/OpenBoard_pt_BR.ts
+++ b/resources/i18n/OpenBoard_pt_BR.ts
@@ -1394,7 +1394,7 @@
         <translation type="vanished">Adicionar à biblioteca</translation>
     </message>
     <message>
-        <source>Object informations</source>
+        <source>Object information</source>
         <translation type="vanished">Informações do Objecto</translation>
     </message>
 </context>
@@ -2014,7 +2014,7 @@
 Por favor, reinicie o aplicativo para ir aos documentos atualizados.</translation>
     </message>
     <message>
-        <source>An error occured during the update. The files have not been affected.</source>
+        <source>An error occurred during the update. The files have not been affected.</source>
         <translation type="vanished">Ocorreu um erro durante a atualização. Os arquivos não foram afetados.</translation>
     </message>
     <message>
--- a/resources/i18n/OpenBoard_ro.ts
+++ b/resources/i18n/OpenBoard_ro.ts
@@ -1263,7 +1263,7 @@
         <translation type="unfinished"></translation>
     </message>
     <message>
-        <source>The name %1 is allready used.
+        <source>The name %1 is already used.
 Keeping this name will replace the document.
 Providing a new name will create a new document.</source>
         <translation type="unfinished"></translation>
@@ -1540,7 +1540,7 @@
         <translation>Adăugare la bibliotecă</translation>
     </message>
     <message>
-        <source>Object informations</source>
+        <source>Object information</source>
         <translation>Informaţii despre obiect</translation>
     </message>
 </context>
@@ -2180,7 +2180,7 @@
 Vă rugăm să reporniţi aplicaţia pentru a accesa documentele actualizate.</translation>
     </message>
     <message>
-        <source>An error occured during the update. The files have not been affected.</source>
+        <source>An error occurred during the update. The files have not been affected.</source>
         <translation>O eroare a intervenit în timpul actualizării. Fişierele nu au fost afectate.</translation>
     </message>
     <message>
--- a/resources/i18n/OpenBoard_ru.ts
+++ b/resources/i18n/OpenBoard_ru.ts
@@ -1263,7 +1263,7 @@
         <translation type="unfinished"></translation>
     </message>
     <message>
-        <source>The name %1 is allready used.
+        <source>The name %1 is already used.
 Keeping this name will replace the document.
 Providing a new name will create a new document.</source>
         <translation type="unfinished"></translation>
@@ -1540,7 +1540,7 @@
         <translation>Добавить в библиотеку</translation>
     </message>
     <message>
-        <source>Object informations</source>
+        <source>Object information</source>
         <translation>Информация об объекте</translation>
     </message>
 </context>
@@ -2180,7 +2180,7 @@
 Перезагрузите приложение, чтобы получить доступ к обновленным документам.</translation>
     </message>
     <message>
-        <source>An error occured during the update. The files have not been affected.</source>
+        <source>An error occurred during the update. The files have not been affected.</source>
         <translation>Во время обновления произошла ошибка. Файлы не изменены.</translation>
     </message>
     <message>
--- a/resources/i18n/OpenBoard_sk.ts
+++ b/resources/i18n/OpenBoard_sk.ts
@@ -1270,7 +1270,7 @@
         <translation type="unfinished"></translation>
     </message>
     <message>
-        <source>The name %1 is allready used.
+        <source>The name %1 is already used.
 Keeping this name will replace the document.
 Providing a new name will create a new document.</source>
         <translation type="unfinished"></translation>
@@ -1574,7 +1574,7 @@
         <translation>Pridať do knižnice</translation>
     </message>
     <message>
-        <source>Object informations</source>
+        <source>Object information</source>
         <translation>Informácie o objekte</translation>
     </message>
 </context>
@@ -2220,7 +2220,7 @@
         <translation>Aktualizovať</translation>
     </message>
     <message>
-        <source>An error occured during the update. The files have not been affected.</source>
+        <source>An error occurred during the update. The files have not been affected.</source>
         <translation>Počas aktualizácie sa vyskytla chyba. Súbory to neovplyvnilo.</translation>
     </message>
     <message>
--- a/resources/i18n/OpenBoard_sv.ts
+++ b/resources/i18n/OpenBoard_sv.ts
@@ -1264,7 +1264,7 @@
         <translation type="unfinished"></translation>
     </message>
     <message>
-        <source>The name %1 is allready used.
+        <source>The name %1 is already used.
 Keeping this name will replace the document.
 Providing a new name will create a new document.</source>
         <translation type="unfinished"></translation>
@@ -1541,7 +1541,7 @@
         <translation>Lägg till bibliotek</translation>
     </message>
     <message>
-        <source>Object informations</source>
+        <source>Object information</source>
         <translation>Objektinformation</translation>
     </message>
 </context>
@@ -2181,7 +2181,7 @@
 Vänligen starta om programmet för att komma åt uppdaterade dokument.</translation>
     </message>
     <message>
-        <source>An error occured during the update. The files have not been affected.</source>
+        <source>An error occurred during the update. The files have not been affected.</source>
         <translation>Ett fel inträffade vid uppdateringen. Filerna har inte påverkats.</translation>
     </message>
     <message>
--- a/resources/i18n/OpenBoard_tr.ts
+++ b/resources/i18n/OpenBoard_tr.ts
@@ -1254,7 +1254,7 @@
         <translation type="unfinished"></translation>
     </message>
     <message>
-        <source>The name %1 is allready used.
+        <source>The name %1 is already used.
 Keeping this name will replace the document.
 Providing a new name will create a new document.</source>
         <translation type="unfinished"></translation>
@@ -1529,7 +1529,7 @@
         <translation>Kütüphaneye ekle</translation>
     </message>
     <message>
-        <source>Object informations</source>
+        <source>Object information</source>
         <translation>Nesne bilgisi</translation>
     </message>
 </context>
@@ -2168,7 +2168,7 @@
 Güncellenen dökümanlara erişmek için uygulamayı yeniden başlatın.</translation>
     </message>
     <message>
-        <source>An error occured during the update. The files have not been affected.</source>
+        <source>An error occurred during the update. The files have not been affected.</source>
         <translation>Güncelleme sırasında bir hata oluştu. Dosyalar değişmedi yada etilenmedi.</translation>
     </message>
     <message>
--- a/resources/i18n/OpenBoard_uk.ts
+++ b/resources/i18n/OpenBoard_uk.ts
@@ -1264,7 +1264,7 @@
         <translation type="unfinished">Відміна</translation>
     </message>
     <message>
-        <source>The name %1 is allready used.
+        <source>The name %1 is already used.
 Keeping this name will replace the document.
 Providing a new name will create a new document.</source>
         <translation type="unfinished"></translation>
@@ -1541,7 +1541,7 @@
         <translation>Додати в бібліотеку</translation>
     </message>
     <message>
-        <source>Object informations</source>
+        <source>Object information</source>
         <translation>Інформація про об&apos;єкт</translation>
     </message>
 </context>
@@ -2189,7 +2189,7 @@
 Перезавантажте програму, щоб отримати доступ до оновлених документів</translation>
     </message>
     <message>
-        <source>An error occured during the update. The files have not been affected.</source>
+        <source>An error occurred during the update. The files have not been affected.</source>
         <translation>Під час оновлення сталася помилка. Файли не змінені.</translation>
     </message>
     <message>
--- a/resources/i18n/OpenBoard_zh.ts
+++ b/resources/i18n/OpenBoard_zh.ts
@@ -1254,7 +1254,7 @@
         <translation type="unfinished"></translation>
     </message>
     <message>
-        <source>The name %1 is allready used.
+        <source>The name %1 is already used.
 Keeping this name will replace the document.
 Providing a new name will create a new document.</source>
         <translation type="unfinished"></translation>
@@ -1528,7 +1528,7 @@
         <translation>添加至图书馆</translation>
     </message>
     <message>
-        <source>Object informations</source>
+        <source>Object information</source>
         <translation>素材信息</translation>
     </message>
 </context>
@@ -2168,7 +2168,7 @@
 请重新启动程序以打开已更新的文件。</translation>
     </message>
     <message>
-        <source>An error occured during the update. The files have not been affected.</source>
+        <source>An error occurred during the update. The files have not been affected.</source>
         <translation>更新过程中发生错误。文件未被改动。</translation>
     </message>
     <message>
--- a/resources/i18n/OpenBoard_zh_CN.ts
+++ b/resources/i18n/OpenBoard_zh_CN.ts
@@ -1254,7 +1254,7 @@
         <translation type="unfinished"></translation>
     </message>
     <message>
-        <source>The name %1 is allready used.
+        <source>The name %1 is already used.
 Keeping this name will replace the document.
 Providing a new name will create a new document.</source>
         <translation type="unfinished"></translation>
@@ -1528,7 +1528,7 @@
         <translation>添加至图书馆</translation>
     </message>
     <message>
-        <source>Object informations</source>
+        <source>Object information</source>
         <translation>素材信息</translation>
     </message>
 </context>
@@ -2168,7 +2168,7 @@
 请重新启动程序以打开已更新的文件。</translation>
     </message>
     <message>
-        <source>An error occured during the update. The files have not been affected.</source>
+        <source>An error occurred during the update. The files have not been affected.</source>
         <translation>更新过程中发生错误。文件未被改动。</translation>
     </message>
     <message>
--- a/resources/i18n/OpenBoard_zh_TW.ts
+++ b/resources/i18n/OpenBoard_zh_TW.ts
@@ -1261,7 +1261,7 @@
         <translation type="unfinished"></translation>
     </message>
     <message>
-        <source>The name %1 is allready used.
+        <source>The name %1 is already used.
 Keeping this name will replace the document.
 Providing a new name will create a new document.</source>
         <translation type="unfinished"></translation>
@@ -1535,7 +1535,7 @@
         <translation>新增至圖書館</translation>
     </message>
     <message>
-        <source>Object informations</source>
+        <source>Object information</source>
         <translation>物件資訊</translation>
     </message>
 </context>
@@ -2162,7 +2162,7 @@
 請重新啟動程式使用該檔案。</translation>
     </message>
     <message>
-        <source>An error occured during the update. The files have not been affected.</source>
+        <source>An error occurred during the update. The files have not been affected.</source>
         <translation>更新過程出現錯誤，原檔案並未更動。</translation>
     </message>
     <message>
--- a/src/adaptors/UBSvgSubsetAdaptor.cpp
+++ b/src/adaptors/UBSvgSubsetAdaptor.cpp
@@ -2083,7 +2083,7 @@
         QFile file(path);
         if (!file.open(QIODevice::WriteOnly))
         {
-            qWarning() << "cannot open file for writing embeded pdf content " << path;
+            qWarning() << "cannot open file for writing embedded pdf content " << path;
             return;
         }
 
--- a/src/api/UBWidgetUniboardAPI.cpp
+++ b/src/api/UBWidgetUniboardAPI.cpp
@@ -574,7 +574,7 @@
         return;
 
     if (!pSuccess) {
-        qDebug() << "can't download the whole data. An error occured";
+        qDebug() << "can't download the whole data. An error occurred";
         return;
     }
 
--- a/src/board/UBFeaturesController.cpp
+++ b/src/board/UBFeaturesController.cpp
@@ -213,7 +213,7 @@
 
 UBFeaturesComputingThread::~UBFeaturesComputingThread()
 {
-//    qDebug() <<  "thread destructor catched";
+//    qDebug() <<  "thread destructor caught";
 
     mMutex.lock();
     abort = true;
@@ -788,7 +788,7 @@
     for (int i = 0; i < curListModel->rowCount(QModelIndex()); i++) {
         QModelIndex ind = curListModel->index(i, 0);
         if (!ind.isValid()) {
-            qDebug() << "incorrect model index catched";
+            qDebug() << "incorrect model index caught";
             continue;
         }
         UBFeature curFeature = curListModel->data(ind, Qt::UserRole + 1).value<UBFeature>();
--- a/src/core/UBDownloadManager.cpp
+++ b/src/core/UBDownloadManager.cpp
@@ -478,7 +478,7 @@
     
     if(NULL != pReply)
     {
-        // Check which error occured:
+        // Check which error occurred:
         switch(pReply->error())
         {
             case QNetworkReply::OperationCanceledError:
@@ -617,7 +617,7 @@
     }
     else
     {
-        // Notify the fact that and error occured during the download
+        // Notify the fact that and error occurred during the download
         emit downloadError(mId);
     }
 }
--- a/src/core/UBPersistenceWorker.cpp
+++ b/src/core/UBPersistenceWorker.cpp
@@ -62,7 +62,7 @@
 
 void UBPersistenceWorker::applicationWillClose()
 {
-    qDebug() << "applicaiton Will close signal received";
+    qDebug() << "application Will close signal received";
     mReceivedApplicationClosing = true;
     mSemaphore.release();
 }
--- a/src/document/UBDocumentController.cpp
+++ b/src/document/UBDocumentController.cpp
@@ -164,7 +164,7 @@
 
 QString UBDocumentReplaceDialog::labelTextWithName(const QString &documentName) const
 {
-    return tr("The name %1 is allready used.\nKeeping this name will replace the document.\nProviding a new name will create a new document.")
+    return tr("The name %1 is already used.\nKeeping this name will replace the document.\nProviding a new name will create a new document.")
             .arg(documentName);
 }
 
@@ -1198,7 +1198,7 @@
 
 void UBDocumentTreeModel::fixNodeName(const QModelIndex &source, const QModelIndex &dest)
 {
-    // Determine whether to provide a name with postfix if the name in current level allready exists
+    // Determine whether to provide a name with postfix if the name in current level already exists
     UBDocumentTreeNode *srcNode = nodeFromIndex(source);
     Q_ASSERT(srcNode);
 
--- a/src/document/UBDocumentController.h
+++ b/src/document/UBDocumentController.h
@@ -248,7 +248,7 @@
 signals:
     void indexChanged(const QModelIndex &newIndex, const QModelIndex &oldIndex);
     void currentIndexMoved(const QModelIndex &newIndex, const QModelIndex &previous); /* Be aware that when you got the signal
-                                                                                       "previous" index would have allready been deleted.
+                                                                                       "previous" index would have already been deleted.
                                                                                         check it for "valid" first */
 
 private:
--- a/src/domain/UBGraphicsGroupContainerItem.cpp
+++ b/src/domain/UBGraphicsGroupContainerItem.cpp
@@ -73,7 +73,7 @@
         return;
     }
 
-    //Check if group is allready rotatable or flippable
+    //Check if group is already rotatable or flippable
     if (childItems().count()) {
         if (UBGraphicsItem::isFlippable(this) && !UBGraphicsItem::isFlippable(item)) {
             Delegate()->setUBFlag(GF_FLIPPABLE_ALL_AXIS, false);
--- a/src/domain/UBGraphicsScene.cpp
+++ b/src/domain/UBGraphicsScene.cpp
@@ -1826,7 +1826,7 @@
         QFile file(completePath);
         if (!file.open(QIODevice::WriteOnly))
         {
-            qWarning() << "cannot open file for writing embeded svg content " << completePath;
+            qWarning() << "cannot open file for writing embedded svg content " << completePath;
             return NULL;
         }
 
--- a/src/gui/UBFeaturesWidget.cpp
+++ b/src/gui/UBFeaturesWidget.cpp
@@ -126,7 +126,7 @@
 void UBFeaturesWidget::currentSelected(const QModelIndex &current)
 {
     if (!current.isValid()) {
-        qWarning() << "SLOT:currentSelected, invalid index catched";
+        qWarning() << "SLOT:currentSelected, invalid index caught";
         return;
     }
 
@@ -904,7 +904,7 @@
 
     mpButtonLayout->addStretch(1);
 
-    mpObjInfoLabel = new QLabel(tr("Object informations"));
+    mpObjInfoLabel = new QLabel(tr("Object information"));
     mpObjInfoLabel->setStyleSheet(QString("color: #888888; font-size : 18px; font-weight:bold;"));
     mpLayout->addWidget(mpObjInfoLabel, 0);
 
@@ -1205,7 +1205,7 @@
     } else if (mimeData->hasUrls()) {
         QList<QUrl> urlList = mimeData->urls();
         foreach (QUrl curUrl, urlList) {
-            qDebug() << "URl catched is " << curUrl.toLocalFile();
+            qDebug() << "URl caught is " << curUrl.toLocalFile();
             curController->moveExternalData(curUrl, parentFeature);
         }
     } else if (mimeData->hasImage()) {
--- a/src/gui/UBUpdateDlg.cpp
+++ b/src/gui/UBUpdateDlg.cpp
@@ -201,7 +201,7 @@
         qsMsg = tr("Files update successful!\nPlease reboot the application to access the updated documents.");
     }
     else {
-        qsMsg = tr("An error occured during the update. The files have not been affected.");
+        qsMsg = tr("An error occurred during the update. The files have not been affected.");
     }
     UBApplication::mainWindow->information(tr("Files update results"), qsMsg);
 }
--- a/src/pdf-merger/FilterPredictor.cpp
+++ b/src/pdf-merger/FilterPredictor.cpp
@@ -264,7 +264,7 @@
 
    if( inSize%(isPNG?_rowLen+1:_rowLen) != 0 )
    {
-      std::cerr<<"Warning : wrong PNG identation inSize "<<inSize<<" rowLen = "<<_rowLen<<" isPNG = "<<isPNG<<"\n";
+      std::cerr<<"Warning : wrong PNG indentation inSize "<<inSize<<" rowLen = "<<_rowLen<<" isPNG = "<<isPNG<<"\n";
       content = out;
       return false;
    }
--- a/src/pdf-merger/Parser.cpp
+++ b/src/pdf-merger/Parser.cpp
@@ -145,7 +145,7 @@
       if( ver < '0' || ver > '4' )
       {
          stringstream errorMsg;
-         errorMsg<<" File with verion 1."<<ver<<" is not currently supported by merge library\n";
+         errorMsg<<" File with version 1."<<ver<<" is not currently supported by merge library\n";
          throw Exception(errorMsg);
       }
    }
--- a/src/podcast/youtube/UBYouTubePublisher.cpp
+++ b/src/podcast/youtube/UBYouTubePublisher.cpp
@@ -155,7 +155,7 @@
     }
     else
     {
-        qDebug() << "Retreived youtube auth token" << mAuthToken;
+        qDebug() << "Retrieved youtube auth token" << mAuthToken;
         postVideoUploadRequest();
     }
 }
