Description: Don't reference AppleScript.
Author: Mike Gabriel <mike.gabriel@das-netzwerkteam.de>

--- a/resources/OpenBoard.qrc
+++ b/resources/OpenBoard.qrc
@@ -240,7 +240,6 @@
         <file>webbrowser/loading.gif</file>
         <file>webbrowser/notfound.html</file>
         <file>webbrowser/object-wrapper.html</file>
-        <file alias="PowerPointImport.applescript">macx/PowerPointImport.applescript</file>
         <file>podcast/OpenBoard.prx</file>
         <file>www/OpenBoard-web-player.html</file>
         <file>images/toolbar/tutorial.png</file>
