diff -ruN openboard.orig/src/desktop/UBDesktopAnnotationController.cpp openboard/src/desktop/UBDesktopAnnotationController.cpp
--- openboard.orig/src/desktop/UBDesktopAnnotationController.cpp	2020-01-23 12:29:35.000000000 +0200
+++ openboard/src/desktop/UBDesktopAnnotationController.cpp	2021-09-07 14:30:00.734900640 +0300
@@ -102,7 +102,7 @@
     mTransparentDrawingView->setScene(mTransparentDrawingScene);
     mTransparentDrawingScene->setDrawingMode(true);
 
-    mDesktopPalette = new UBDesktopPalette(mTransparentDrawingView, rightPalette); 
+    mDesktopPalette = new UBDesktopPalette(mTransparentDrawingView, rightPalette);
     // This was not fix, parent reverted
     // FIX #633: The palette must be 'floating' in order to stay on top of the library palette
 
@@ -903,6 +903,10 @@
         }
 
 #ifdef Q_OS_LINUX
+        // workaround required for Ubuntu 18 (issue : drawing is not visible in Desktop Mode until updateMask is recalled)
+        QRect desktopRect = QApplication::desktop()->screenGeometry(mDesktopPalette->pos());
+        p.drawRect(desktopRect);
+
         //Rquiered only for compiz wm
         //TODO. Window manager detection screen
 
