From 9ab048e82be68ea2b9d969209444bfd0433ca85d Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Tuomas=20R=C3=A4s=C3=A4nen?= <tuomas.rasanen@opinsys.fi>
Date: Fri, 16 Feb 2024 14:19:35 +0200
Subject: [PATCH 2/4] winegstreamer: add wg_format_from_caps_video_h264()

---
 dlls/winegstreamer/wg_format.c | 128 +++++++++++++++++++++++++++++++++
 1 file changed, 128 insertions(+)

diff --git a/dlls/winegstreamer/wg_format.c b/dlls/winegstreamer/wg_format.c
index 6da97685736..d4e0a457729 100644
--- a/dlls/winegstreamer/wg_format.c
+++ b/dlls/winegstreamer/wg_format.c
@@ -390,6 +390,129 @@ static void wg_format_from_caps_video_mpeg1(struct wg_format *format, const GstC
     format->u.video_mpeg1.fps_d = fps_d;
 }
 
+
+static void wg_format_from_caps_video_h264(struct wg_format *format, const GstCaps *caps)
+{
+    const GstStructure *structure = gst_caps_get_structure(caps, 0);
+    int32_t width, height;
+    uint32_t fps_n, fps_d, profile, level, codec_data_len;
+    const gchar *profile_str = NULL;
+    const gchar *level_str = NULL;
+    const GValue *codec_data_value;
+    GstBuffer *codec_data;
+    GstMapInfo map;
+
+    if (!gst_structure_get_int(structure, "width", &width))
+    {
+        GST_WARNING("Missing \"width\" value in %" GST_PTR_FORMAT ".", caps);
+        return;
+    }
+    if (!gst_structure_get_int(structure, "height", &height))
+    {
+        GST_WARNING("Missing \"height\" value in %" GST_PTR_FORMAT ".", caps);
+        return;
+    }
+    if (!gst_structure_get_fraction(structure, "framerate", &fps_n, &fps_d))
+    {
+        GST_WARNING("Missing \"framerate\" value in %" GST_PTR_FORMAT ".", caps);
+        fps_n = 0;
+        fps_d = 1;
+    }
+    if (!(profile_str = gst_structure_get_string(structure, "profile")))
+    {
+        GST_WARNING("Missing \"profile\" value in %" GST_PTR_FORMAT ".", caps);
+        return; // This should always exist, because the fallack value is "baseline".
+    }
+
+    if (!strcmp(profile_str, "main"))
+        profile = eAVEncH264VProfile_Main;
+    else if (!strcmp(profile_str, "high"))
+        profile = eAVEncH264VProfile_High;
+    else if (!strcmp(profile_str, "high-4:4:4"))
+        profile = eAVEncH264VProfile_444;
+    else if (!strcmp(profile_str, "baseline"))
+        profile = eAVEncH264VProfile_unknown;
+    else
+    {
+        GST_FIXME("Missing H264 profile_str -> profile conversion for '%s'", profile_str);
+        // FIXME: wg_format_to_caps_video_h264() function sets
+        // unexpected profile_str. So, someone has added support for
+        // new profile, but not added it here.
+        return;
+    }
+
+    level = 0;
+    if ((level_str = gst_structure_get_string(structure, "level")))
+    {
+        if (strcmp(level_str, "1"))
+            level = eAVEncH264VLevel1;
+        else if (strcmp(level_str, "1.1"))
+            level = eAVEncH264VLevel1_1;
+        else if (strcmp(level_str, "1.2"))
+            level = eAVEncH264VLevel1_2;
+        else if (strcmp(level_str, "1.3"))
+            level = eAVEncH264VLevel1_3;
+        else if (strcmp(level_str, "2"))
+            level = eAVEncH264VLevel2;
+        else if (strcmp(level_str, "2.1"))
+            level = eAVEncH264VLevel2_1;
+        else if (strcmp(level_str, "2.2"))
+            level = eAVEncH264VLevel2_2;
+        else if (strcmp(level_str, "3"))
+            level = eAVEncH264VLevel3;
+        else if (strcmp(level_str, "3.1"))
+            level = eAVEncH264VLevel3_1;
+        else if (strcmp(level_str, "3.2"))
+            level = eAVEncH264VLevel3_2;
+        else if (strcmp(level_str, "4"))
+            level = eAVEncH264VLevel4;
+        else if (strcmp(level_str, "4.1"))
+            level = eAVEncH264VLevel4_1;
+        else if (strcmp(level_str, "4.2"))
+            level = eAVEncH264VLevel4_2;
+        else if (strcmp(level_str, "5"))
+            level = eAVEncH264VLevel5;
+        else if (strcmp(level_str, "5.1"))
+            level = eAVEncH264VLevel5_1;
+        else if (strcmp(level_str, "5.2"))
+            level = eAVEncH264VLevel5_2;
+        else
+        {
+            GST_FIXME("Missing H264 level_str -> level conversion for '%s'", level_str);
+            // FIXME: wg_format_to_caps_video_h264() function sets
+            // unexpected level_str. So, someone has added support for
+            // new level, but not added it here.
+            return;
+        }
+    }
+
+    codec_data_len = 0;
+    if ((codec_data_value = gst_structure_get_value(structure, "streamheader")) && (codec_data = gst_value_get_buffer(codec_data_value)))
+    {
+        gst_buffer_map(codec_data, &map, GST_MAP_READ);
+        if (map.size <= sizeof(format->u.video_h264.codec_data))
+        {
+            codec_data_len = map.size;
+            memcpy(format->u.video_h264.codec_data, map.data, map.size);
+        }
+        else
+            GST_WARNING("Too big codec_data value (%u) in %" GST_PTR_FORMAT ".", (UINT)map.size, caps);
+        gst_buffer_unmap(codec_data, &map);
+    }
+    else
+        GST_WARNING("Failed to get \"streamheader\" value from %" GST_PTR_FORMAT ".", caps);
+
+    format->major_type = WG_MAJOR_TYPE_VIDEO_H264;
+    format->u.video_h264.width = width;
+    format->u.video_h264.height = height;
+    format->u.video_h264.fps_n = fps_n;
+    format->u.video_h264.fps_d = fps_d;
+    format->u.video_h264.profile = profile;
+    format->u.video_h264.level = level;
+    format->u.video_h264.codec_data_len = codec_data_len;
+}
+
+
 void wg_format_from_caps(struct wg_format *format, const GstCaps *caps)
 {
     const GstStructure *structure = gst_caps_get_structure(caps, 0);
@@ -432,6 +555,10 @@ void wg_format_from_caps(struct wg_format *format, const GstCaps *caps)
     {
         wg_format_from_caps_video_mpeg1(format, caps);
     }
+    else if (!strcmp(name, "video/x-h264") && gst_structure_get_boolean(structure, "parsed", &parsed) && parsed)
+    {
+        wg_format_from_caps_video_h264(format, caps);
+    }
     else
     {
         GST_FIXME("Unhandled caps %" GST_PTR_FORMAT ".", caps);
@@ -743,6 +870,7 @@ static GstCaps *wg_format_to_caps_video_h264(const struct wg_format *format)
         gst_caps_set_simple(caps, "streamheader", GST_TYPE_BUFFER, buffer, NULL);
         gst_buffer_unref(buffer);
     }
+    gst_caps_set_simple(caps, "parsed", G_TYPE_BOOLEAN, TRUE, NULL);
 
     return caps;
 }
-- 
2.39.2

