From 15db8fa6a915a95f66a0a93c6a256485f160824d Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Tuomas=20R=C3=A4s=C3=A4nen?= <tuomas.rasanen@opinsys.fi>
Date: Fri, 16 Feb 2024 19:56:06 +0200
Subject: [PATCH 4/4] constrained+baseline etc.

---
 dlls/winegstreamer/wg_format.c | 12 +++++++++---
 1 file changed, 9 insertions(+), 3 deletions(-)

diff --git a/dlls/winegstreamer/wg_format.c b/dlls/winegstreamer/wg_format.c
index dbb8abb6fa1..2ce1aca17de 100644
--- a/dlls/winegstreamer/wg_format.c
+++ b/dlls/winegstreamer/wg_format.c
@@ -430,8 +430,10 @@ static void wg_format_from_caps_video_h264(struct wg_format *format, const GstCa
         profile = eAVEncH264VProfile_High;
     else if (!strcmp(profile_str, "high-4:4:4"))
         profile = eAVEncH264VProfile_444;
+    else if (!strcmp(profile_str, "constrained-baseline"))
+        profile = eAVEncH264VProfile_ConstrainedBase;
     else if (!strcmp(profile_str, "baseline"))
-        profile = eAVEncH264VProfile_unknown;
+        profile = eAVEncH264VProfile_Base;
     else
     {
         GST_FIXME("Missing H264 profile_str -> profile conversion for '%s'", profile_str);
@@ -491,6 +493,7 @@ static void wg_format_from_caps_video_h264(struct wg_format *format, const GstCa
     {
         GST_WARNING("Found \"codec_data\" value from %" GST_PTR_FORMAT ".", caps);
         gst_buffer_map(codec_data, &map, GST_MAP_READ);
+        GST_WARNING("tjjr: map size %d", map.size);
         if (map.size <= sizeof(format->u.video_h264.codec_data))
         {
             codec_data_len = map.size;
@@ -536,7 +539,7 @@ void wg_format_from_caps(struct wg_format *format, const GstCaps *caps)
         if (gst_video_info_from_caps(&info, caps))
             wg_format_from_video_info(format, &info);
     }
-    else if (!strcmp(name, "audio/mpeg") && gst_structure_get_boolean(structure, "parsed", &parsed) && parsed)
+    else if (!strcmp(name, "audio/mpeg"))
     {
         wg_format_from_caps_audio_mpeg1(format, caps);
     }
@@ -556,7 +559,7 @@ void wg_format_from_caps(struct wg_format *format, const GstCaps *caps)
     {
         wg_format_from_caps_video_mpeg1(format, caps);
     }
-    else if (!strcmp(name, "video/x-h264") && gst_structure_get_boolean(structure, "parsed", &parsed) && parsed)
+    else if (!strcmp(name, "video/x-h264"))
     {
         wg_format_from_caps_video_h264(format, caps);
     }
@@ -820,10 +823,12 @@ static GstCaps *wg_format_to_caps_video_h264(const struct wg_format *format)
         case eAVEncH264VProfile_Main: profile = "main"; break;
         case eAVEncH264VProfile_High: profile = "high"; break;
         case eAVEncH264VProfile_444:  profile = "high-4:4:4"; break;
+        case eAVEncH264VProfile_ConstrainedBase: profile = "constrained-baseline"; break;
         default:
             GST_FIXME("H264 profile attribute %u not implemented.", format->u.video_h264.profile);
             /* fallthrough */
         case eAVEncH264VProfile_unknown:
+            GST_WARNING("H264 profile is unknown, fallback to baseline.");
             profile = "baseline";
             break;
     }
@@ -859,6 +864,7 @@ static GstCaps *wg_format_to_caps_video_h264(const struct wg_format *format)
 
     if (format->u.video_h264.codec_data_len)
     {
+        GST_INFO("tjjr: code data len: %d", format->u.video_h264.codec_data_len);
         if (!(buffer = gst_buffer_new_and_alloc(format->u.video_h264.codec_data_len)))
         {
             gst_caps_unref(caps);
-- 
2.39.2

