diff -ruN gtk4/modules/printbackends/gtkprintbackendcups.c gtk4/modules/printbackends/gtkprintbackendcups.c
--- gtk4/modules/printbackends/gtkprintbackendcups.c	2022-12-22 20:44:48.000000000 +0200
+++ gtk4/modules/printbackends/gtkprintbackendcups.c	2023-11-28 12:00:45.427666107 +0200
@@ -3942,6 +3942,7 @@
 cups_get_printer_list (GtkPrintBackend *backend)
 {
   GtkPrintBackendCups *cups_backend;
+  char *show_avahi_printers;
 
   cups_backend = GTK_PRINT_BACKEND_CUPS (backend);
 
@@ -3956,7 +3957,10 @@
           g_source_set_name_by_id (cups_backend->list_printers_poll, "[gtk] cups_request_printer_list");
         }
 
-      avahi_request_printer_list (cups_backend);
+      show_avahi_printers = g_getenv("GTK_SHOW_AVAHI_PRINTERS");
+      if (!(show_avahi_printers != NULL && strcmp(show_avahi_printers, "false") == 0)) {
+        avahi_request_printer_list (cups_backend);
+      }
     }
 }
 
