From 554dff1927176579d652f2fe60c90e9abbad4c65 Mon Sep 17 00:00:00 2001
From: Howard Chu <hyc@openldap.org>
Date: Mon, 30 Nov 2020 16:20:18 +0000
Subject: [PATCH] ITS#9409 saslauthz: use slap_sl_free in prev commit

---
 servers/slapd/saslauthz.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/servers/slapd/saslauthz.c b/servers/slapd/saslauthz.c
index cc5a292de7..4a9420b37c 100644
--- a/servers/slapd/saslauthz.c
+++ b/servers/slapd/saslauthz.c
@@ -860,7 +860,7 @@ done:
 
 	if ( lud_dn ) {
 		if ( ludp->lud_dn != lud_dn ) {
-			ch_free( ludp->lud_dn );
+			slap_sl_free( ludp->lud_dn, ctx );
 		}
 		ludp->lud_dn = lud_dn;
 	}
-- 
2.20.1

