From c0b61a9486508e5202aa2e0cfb68c9813731b439 Mon Sep 17 00:00:00 2001
From: Howard Chu <hyc@openldap.org>
Date: Mon, 30 Nov 2020 11:45:46 +0000
Subject: [PATCH 2/2] ITS#9409 saslauthz: use ch_free on normalized DN

---
 servers/slapd/saslauthz.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/servers/slapd/saslauthz.c b/servers/slapd/saslauthz.c
index 982fe3120d..cc5a292de7 100644
--- a/servers/slapd/saslauthz.c
+++ b/servers/slapd/saslauthz.c
@@ -860,7 +860,7 @@ done:
 
 	if ( lud_dn ) {
 		if ( ludp->lud_dn != lud_dn ) {
-			ber_memfree( ludp->lud_dn );
+			ch_free( ludp->lud_dn );
 		}
 		ludp->lud_dn = lud_dn;
 	}
-- 
2.20.1

