From 5a2017d4e61a6ddc4dcb4415028e0d08eb6bca26 Mon Sep 17 00:00:00 2001
From: Howard Chu <hyc@openldap.org>
Date: Tue, 1 Dec 2020 18:32:35 +0000
Subject: [PATCH] ITS#9412 fix AVA_Sort on invalid RDN

---
 servers/slapd/dn.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/servers/slapd/dn.c b/servers/slapd/dn.c
index 06698b089e..7a095ba9e9 100644
--- a/servers/slapd/dn.c
+++ b/servers/slapd/dn.c
@@ -233,6 +233,7 @@ AVA_Sort( LDAPRDN rdn, int nAVAs )
 {
 	LDAPAVA	*ava_i;
 	int		i;
+	int		rc = LDAP_SUCCESS;
 
 	assert( rdn != NULL );
 
@@ -250,7 +251,7 @@ AVA_Sort( LDAPRDN rdn, int nAVAs )
 			/* RFC4512 does not allow multiple AVAs
 			 * with the same attribute type in RDN (ITS#5968) */
 			if ( a == 0 )
-				return LDAP_INVALID_DN_SYNTAX;
+				rc = LDAP_INVALID_DN_SYNTAX;
 
 			if ( a > 0 )
 				break;
@@ -259,7 +260,7 @@ AVA_Sort( LDAPRDN rdn, int nAVAs )
 		}
 		rdn[ j+1 ] = ava_i;
 	}
-	return LDAP_SUCCESS;
+	return rc;
 }
 
 static int
-- 
2.20.1

