description: don't build fonts in the arch:all build
author: Michael Gilbert <mgilbert@debian.org>

--- a/configure.ac
+++ b/configure.ac
@@ -3894,7 +3894,6 @@ WINE_CONFIG_MAKEFILE(dlls/xmllite/tests)
 WINE_CONFIG_MAKEFILE(dlls/xolehlp)
 WINE_CONFIG_MAKEFILE(dlls/xpsprint)
 WINE_CONFIG_MAKEFILE(dlls/xpssvcs)
-WINE_CONFIG_MAKEFILE(fonts)
 WINE_CONFIG_MAKEFILE(include)
 WINE_CONFIG_MAKEFILE(libs/port)
 WINE_CONFIG_MAKEFILE(libs/wine)
@@ -4006,7 +4005,6 @@ WINE_CONFIG_MAKEFILE(programs/xcopy)
 WINE_CONFIG_MAKEFILE(programs/xcopy/tests)
 WINE_CONFIG_MAKEFILE(server)
 WINE_CONFIG_MAKEFILE(tools,,[test "x$enable_tools" = xno])
-WINE_CONFIG_MAKEFILE(tools/sfnt2fon,,[test "x$enable_tools" = xno])
 WINE_CONFIG_MAKEFILE(tools/widl,,[test "x$enable_tools" = xno])
 WINE_CONFIG_MAKEFILE(tools/winebuild,,[test "x$enable_tools" = xno])
 WINE_CONFIG_MAKEFILE(tools/winedump,,[test "x$enable_tools" = xno])
--- a/tools/sfnt2fon/Makefile.in
+++ b/tools/sfnt2fon/Makefile.in
@@ -3,4 +3,4 @@ PROGRAMS = sfnt2fon
 C_SRCS = sfnt2fon.c
 
 EXTRAINCL = $(FREETYPE_CFLAGS)
-EXTRALIBS = $(FREETYPE_LIBS)
+EXTRALIBS = $(FREETYPE_LIBS) -L../../libs/port
