diff -ruN gnome-shell-3.30.2.build-3/js/ui/sessionMode.js gnome-shell-3.30.2.build-4/js/ui/sessionMode.js
--- gnome-shell-3.30.2.build-3/js/ui/sessionMode.js	2019-04-17 12:34:15.649040481 +0300
+++ gnome-shell-3.30.2.build-4/js/ui/sessionMode.js	2019-04-17 12:44:39.076357325 +0300
@@ -46,7 +46,9 @@
         isGreeter: true,
         isPrimary: true,
         unlockDialog: imports.gdm.loginDialog.LoginDialog,
-        components: ['polkitAgent'],
+        components: Config.HAVE_NETWORKMANAGER ?
+                    ['networkAgent', 'polkitAgent'] :
+                    ['polkitAgent'],
         panel: {
             left: [],
             center: ['dateMenu'],
diff -ruN gnome-shell-3.30.2.build-3/js/ui/status/network.js gnome-shell-3.30.2.build-4/js/ui/status/network.js
--- gnome-shell-3.30.2.build-3/js/ui/status/network.js	2018-11-14 00:50:31.000000000 +0200
+++ gnome-shell-3.30.2.build-4/js/ui/status/network.js	2019-04-17 12:45:11.472940702 +0300
@@ -1681,7 +1681,7 @@
     },
 
     _sessionUpdated() {
-        let sensitive = !Main.sessionMode.isLocked && !Main.sessionMode.isGreeter;
+        let sensitive = !Main.sessionMode.isLocked;
         this.menu.setSensitive(sensitive);
     },
 
