diff -ruN gnome-shell-3.30.2/js/gdm/loginDialog.js gnome-shell-3.30.2/js/gdm/loginDialog.js
--- gnome-shell-3.30.2/js/gdm/loginDialog.js	2019-04-17 10:37:49.000000000 +0300
+++ gnome-shell-3.30.2/js/gdm/loginDialog.js	2019-04-17 10:39:54.111002882 +0300
@@ -164,7 +164,7 @@
 var UserList = new Lang.Class({
     Name: 'UserList',
 
-    _init() {
+    _init(accepted_users) {
         this.actor = new St.ScrollView({ style_class: 'login-dialog-user-list-view'});
         this.actor.set_policy(Gtk.PolicyType.NEVER,
                               Gtk.PolicyType.AUTOMATIC);
@@ -176,6 +176,8 @@
         this.actor.add_actor(this._box);
         this._items = {};
 
+        this._accepted_users = accepted_users.split(',');
+
         this.actor.connect('key-focus-in', this._moveFocusToItems.bind(this));
     },
 
@@ -261,6 +263,17 @@
         if (user.locked)
            return;
 
+        is_accepted = false;
+        for (let i = 0; i < this._accepted_users.length; i++) {
+            if (this._accepted_users[i] == '*' ||
+              this._accepted_users[i] == user.user_name) {
+                is_accepted = true;
+                break;
+            }
+        }
+        if (!is_accepted)
+            return;
+
         let userName = user.get_user_name();
 
         if (!userName)
@@ -445,7 +458,9 @@
                                                     visible: false });
         this.actor.add_child(this._userSelectionBox);
 
-        this._userList = new UserList();
+        let accepted_users = this._settings.get_string('accepted-usernames-for-userlist');
+        this._userList = new UserList(accepted_users);
+
         this._userSelectionBox.add(this._userList.actor,
                                    { expand: true,
                                      x_fill: true,
