diff -ruN gnome-shell.0/js/gdm/loginDialog.js gnome-shell.1/js/gdm/loginDialog.js
--- gnome-shell.0/js/gdm/loginDialog.js	2021-10-04 10:16:24.588968776 +0300
+++ gnome-shell.1/js/gdm/loginDialog.js	2021-10-04 10:36:08.208741263 +0300
@@ -160,7 +160,7 @@
         'item-added': { param_types: [UserListItem.$gtype] },
     },
 }, class UserList extends St.ScrollView {
-    _init() {
+    _init(accepted_users) {
         super._init({
             style_class: 'login-dialog-user-list-view',
             x_expand: true,
@@ -175,6 +175,8 @@
 
         this.add_actor(this._box);
         this._items = {};
+
+        this._accepted_users = accepted_users.split(',');
     }
 
     vfunc_key_focus_in() {
@@ -268,6 +270,17 @@
 
         this.removeUser(user);
 
+        let is_accepted = false;
+        for (let i = 0; i < this._accepted_users.length; i++) {
+            if (this._accepted_users[i] == '*' ||
+              this._accepted_users[i] == userName) {
+                is_accepted = true;
+                break;
+            }
+        }
+        if (!is_accepted)
+            return;
+
         let item = new UserListItem(user);
         this._box.add_child(item);
 
@@ -442,7 +455,8 @@
                                                     visible: false });
         this.add_child(this._userSelectionBox);
 
-        this._userList = new UserList();
+        let accepted_users = this._settings.get_string('accepted-usernames-for-userlist');
+        this._userList = new UserList(accepted_users);
         this._userSelectionBox.add_child(this._userList);
 
         this._authPrompt = new AuthPrompt.AuthPrompt(this._gdmClient, AuthPrompt.AuthPromptMode.UNLOCK_OR_LOG_IN);
