diff -ruN gnome-shell.orig/js/ui/endSessionDialog.js gnome-shell/js/ui/endSessionDialog.js
--- gnome-shell.orig/js/ui/endSessionDialog.js	2018-09-25 13:18:57.706902585 +0300
+++ gnome-shell/js/ui/endSessionDialog.js	2018-09-25 13:19:11.039160033 +0300
@@ -90,7 +90,9 @@
     },
     checkBoxText: C_("checkbox", "Install pending software updates"),
     showBatteryWarning: true,
-    confirmButtons: [{ signal: 'ConfirmedReboot',
+    confirmButtons: [{ signal: 'ConfirmedLogout',
+                       label:  C_("button", "Log Out") },
+                     { signal: 'ConfirmedReboot',
                        label:  C_("button", "Restart") },
                      { signal: 'ConfirmedShutdown',
                        label:  C_("button", "Power Off") }],
