diff -ruN gnome-shell.0/js/gdm/authPrompt.js gnome-shell.1/js/gdm/authPrompt.js
--- gnome-shell.0/js/gdm/authPrompt.js	2021-10-05 08:19:46.000000000 +0300
+++ gnome-shell.1/js/gdm/authPrompt.js	2021-10-05 08:22:44.545997815 +0300
@@ -270,6 +270,8 @@
         ShellEntry.addContextMenu(this._textEntry, { actionMode: Shell.ActionMode.NONE });
 
         this._passwordEntry = new St.PasswordEntry(entryParams);
+        // Do not allow password peeking in Puavo OS.
+        this._passwordEntry.set_show_peek_icon(false);
         ShellEntry.addContextMenu(this._passwordEntry, { actionMode: Shell.ActionMode.NONE });
 
         this._entry = this._passwordEntry;
diff -ruN gnome-shell.0/js/ui/shellEntry.js gnome-shell.1/js/ui/shellEntry.js
--- gnome-shell.0/js/ui/shellEntry.js	2021-03-15 23:28:45.035283600 +0200
+++ gnome-shell.1/js/ui/shellEntry.js	2021-10-05 08:27:36.137814613 +0300
@@ -36,8 +36,14 @@
 
     _makePasswordItem() {
         let item = new PopupMenu.PopupMenuItem('');
-        item.connect('activate', this._onPasswordActivated.bind(this));
-        this.addMenuItem(item);
+        // Do not add this item to menu or active it ever!
+        // This is a design mistake, user passwords end up showing accidentally
+        // sometimes.  For example, if user writes username+password and
+        // presses enter three times after that in a fast succession, the
+        // password will appear on the screen before the login proceeds
+        // to the desktop.
+        // item.connect('activate', this._onPasswordActivated.bind(this));
+        // this.addMenuItem(item);
         this._passwordItem = item;
     }
 
@@ -92,7 +98,9 @@
     }
 
     _onPasswordActivated() {
-        this._entry.password_visible  = !this._entry.password_visible;
+        // Password should never be visible in Puavo OS.
+        // this._entry.password_visible  = !this._entry.password_visible;
+        this._entry.password_visible = false;
     }
 };
 
