diff -ruN gnome-shell/js/ui/shellEntry.js gnome-shell/js/ui/shellEntry.js
--- gnome-shell/js/ui/shellEntry.js	2021-03-15 23:28:45.035283600 +0200
+++ gnome-shell/js/ui/shellEntry.js	2021-05-17 15:13:42.116611359 +0300
@@ -36,8 +36,14 @@
 
     _makePasswordItem() {
         let item = new PopupMenu.PopupMenuItem('');
-        item.connect('activate', this._onPasswordActivated.bind(this));
-        this.addMenuItem(item);
+        // Do not add this item to menu or active it ever!
+        // This is a design mistake, user passwords end up showing accidentally
+        // sometimes.  For example, if user writes username+password and
+        // presses enter three times after that in a fast succession, the
+        // password will appear on the screen before the login proceeds
+        // to the desktop.
+        // item.connect('activate', this._onPasswordActivated.bind(this));
+        // this.addMenuItem(item);
         this._passwordItem = item;
     }
 
