diff -ruN gnome-shell/js/ui/shellEntry.js gnome-shell.1/js/ui/shellEntry.js
--- gnome-shell/js/ui/shellEntry.js	2016-07-29 07:43:45.000000000 +0300
+++ gnome-shell/js/ui/shellEntry.js	2019-05-14 10:56:46.387440202 +0300
@@ -40,9 +40,15 @@
 
     _makePasswordItem: function() {
         let item = new PopupMenu.PopupMenuItem('');
-        item.connect('activate', Lang.bind(this,
-                                           this._onPasswordActivated));
-        this.addMenuItem(item);
+        // Do not add this item to menu or active it ever!
+        // This is a design mistake, user passwords end up showing accidentally
+        // sometimes.  For example, if user writes username+password and
+        // presses enter three times after that in a fast succession, the
+        // password will appear on the screen before the login proceeds
+        // to the desktop.
+        // item.connect('activate', Lang.bind(this,
+        //                                    this._onPasswordActivated));
+        // this.addMenuItem(item);
         this._passwordItem = item;
     },
 
