diff -ruN gnome-shell.orig/js/ui/shellEntry.js gnome-shell/js/ui/shellEntry.js
--- gnome-shell.orig/js/ui/shellEntry.js	2019-07-01 12:07:31.682495482 +0300
+++ gnome-shell/js/ui/shellEntry.js	2019-07-01 12:25:10.412436811 +0300
@@ -41,8 +41,14 @@
 
     _makePasswordItem() {
         let item = new PopupMenu.PopupMenuItem('');
-        item.connect('activate', this._onPasswordActivated.bind(this));
-        this.addMenuItem(item);
+        // Do not add this item to menu or active it ever!
+        // This is a design mistake, user passwords end up showing accidentally
+        // sometimes.  For example, if user writes username+password and
+        // presses enter three times after that in a fast succession, the
+        // password will appear on the screen before the login proceeds
+        // to the desktop.
+        // item.connect('activate', this._onPasswordActivated.bind(this));
+        // this.addMenuItem(item);
         this._passwordItem = item;
     },
 
