diff -ruN gnome-shell-3.30.2.build-4/js/gdm/loginDialog.js gnome-shell-3.30.2.build-5/js/gdm/loginDialog.js
--- gnome-shell-3.30.2.build-4/js/gdm/loginDialog.js	2019-04-17 12:00:25.946460901 +0300
+++ gnome-shell-3.30.2.build-5/js/gdm/loginDialog.js	2019-04-17 12:50:31.497279528 +0300
@@ -865,6 +865,8 @@
         } else {
             this._hideUserListAndBeginVerification();
         }
+
+        GLib.spawn_command_line_async('/usr/bin/puavo-autopilot-login');
     },
 
     _onDefaultSessionChanged(client, sessionId) {
