diff -ruN gnome-shell-3.30.2.orig/data/theme/gnome-shell-sass/_common.scss gnome-shell-3.30.2/data/theme/gnome-shell-sass/_common.scss
--- gnome-shell-3.30.2.orig/data/theme/gnome-shell-sass/_common.scss	2019-04-17 13:31:18.973752260 +0300
+++ gnome-shell-3.30.2/data/theme/gnome-shell-sass/_common.scss	2019-04-17 13:32:07.150639982 +0300
@@ -2020,3 +2020,12 @@
     border-radius: 4px;
     padding: 6px;
   }
+
+// Puavo customizations
+.destroy-user-session {
+  color: #cc8800;
+  font-size: 70%;
+  font-weight: bold;
+  margin-top: 3em; }
+  .destroy-user-session:hover {
+    color: #ffaa00; }
diff -ruN gnome-shell-3.30.2.orig/js/gdm/authPrompt.js gnome-shell-3.30.2/js/gdm/authPrompt.js
--- gnome-shell-3.30.2.orig/js/gdm/authPrompt.js	2019-04-17 13:31:18.973752260 +0300
+++ gnome-shell-3.30.2/js/gdm/authPrompt.js	2019-04-17 13:32:19.018858356 +0300
@@ -2,10 +2,12 @@
 
 const Clutter = imports.gi.Clutter;
 const Gio = imports.gi.Gio;
+const GLib = imports.gi.GLib;
 const Lang = imports.lang;
 const Pango = imports.gi.Pango;
 const Signals = imports.signals;
 const St = imports.gi.St;
+const Util = imports.misc.util;
 
 const Animation = imports.ui.animation;
 const Batch = imports.gdm.batch;
@@ -39,6 +41,12 @@
     DONT_PROVIDE_USERNAME: 1
 };
 
+// will throw exceptions, remember to handle them
+function readTextFile(name)
+{
+    return Shell.get_file_contents_utf8_sync(name);
+}
+
 var AuthPrompt = new Lang.Class({
     Name: 'AuthPrompt',
 
@@ -48,6 +56,13 @@
         this._gdmClient = gdmClient;
         this._mode = mode;
 
+        gio_settings = new Gio.Settings({ schema_id: GdmUtil.LOGIN_SCREEN_SCHEMA });
+        if (gio_settings.get_boolean(GdmUtil.GUESTUSER_KEY)) {
+          this._guestuser_login = 'guest';
+        } else {
+          this._guestuser_login = '';
+        }
+
         let reauthenticationOnly;
         if (this._mode == AuthPromptMode.UNLOCK_ONLY)
             reauthenticationOnly = true;
@@ -68,6 +83,14 @@
         this.connect('next', () => {
                 this.updateSensitivity(false);
                 this.startSpinning();
+
+                if (this._userName && !this._userNameEntered) {
+                        if (!this._entry.menu.isPassword) {    // don't show passwords!
+                                this._userName.text = this._entry.text;
+                                this._userNameEntered = true;
+                        }
+                }
+
                 if (this._queryingService) {
                     this._userVerifier.answerQuery(this._queryingService, this._entry.text);
                 } else {
@@ -75,6 +98,11 @@
                 }
             });
 
+        this.connect('guest', Lang.bind(this, function() {
+                         this._entry.text = this._guestuser_login;
+                         this.emit('next');
+                     }));
+
         this.actor = new St.BoxLayout({ style_class: 'login-dialog-prompt-layout',
                                         vertical: true });
         this.actor.connect('destroy', this._onDestroy.bind(this));
@@ -91,6 +119,25 @@
                          x_fill: true,
                          y_fill: true,
                          expand: true });
+
+        if (this._mode == AuthPromptMode.UNLOCK_OR_LOG_IN) {
+            // Display the username above the password prompt, but
+            // ONLY in the login screen, not in the unlocking screen.
+            // The label is always visible, it's just empty before
+            // the name has been entered.
+            this._userName = new St.Label({ style_class:
+                'login-dialog-username', text: ' ' });
+
+            this.actor.add(this._userName, {
+                expand: true,
+                x_fill: false,
+                y_fill: true,
+                x_align: St.Align.MIDDLE });
+                this._userName.show();
+        }
+
+        this._userNameEntered = false;
+
         this._label = new St.Label({ style_class: 'login-dialog-prompt-label' });
 
         this.actor.add(this._label,
@@ -123,6 +170,74 @@
                          x_align: St.Align.MIDDLE,
                          y_align: St.Align.END });
 
+        this.guestButton = new St.Button({ style_class: 'modal-dialog-button button login-dialog-guest-button',
+                                           button_mask: St.ButtonMask.ONE | St.ButtonMask.THREE,
+                                           reactive: true,
+                                           can_focus: true,
+                                           label: _("Login as guest") });
+        this.guestButton.connect('clicked',
+                                 Lang.bind(this, function() { this.emit('guest'); }));
+        this.actor.add(this.guestButton,
+                       { expand: true,
+                         x_fill: false,
+                         y_fill: true,
+                         x_align: St.Align.MIDDLE });
+
+        if (reauthenticationOnly) {
+            // hide the guest login button on the unlock screen
+            this.guestButton.hide();
+        } else {
+            // initially the guest button is clickable
+            this._updateGuestButtonSensitivity(true);
+        }
+
+        if (reauthenticationOnly) {
+            // Show the machine hostname, host type and release name in the lock screen.
+            // I tried to create a separate label element for this in loginDialog.js, but
+            // I just couldn't get it to work (the label never shows up). So we'll do it
+            // here. The font size has been set to small to make long hostnames visible
+            // (if the hostname is too long, it gets ellipsized). These things should be
+            // done using extensions, but I couldn't find a way to run extensions in the
+            // lock screen.
+            var hostType,
+                hostName,
+                releaseName;
+
+            try { hostType = readTextFile('/etc/puavo/hosttype').trim(); } catch (e) { hostType = '?'; }
+            try { hostName = readTextFile('/etc/puavo/hostname').trim(); } catch (e) { hostName = '?'; }
+            try { releaseName = readTextFile('/etc/puavo-image/release').trim(); } catch (e) { releaseName = '?'; }
+
+            this._machineInfoText = new St.Label({
+                text: hostType + ' | ' + releaseName + ' | ' + hostName,
+                style: 'font-size: 80%; text-align: center; margin-top: 20px;'
+            });
+
+            this.actor.add_child(this._machineInfoText);
+            this._machineInfoText.show();
+            // This is a hack from start to end, but right now it does the job.
+        }
+
+        if (reauthenticationOnly) {
+            let button_opts = {
+              button_mask: St.ButtonMask.ONE | St.ButtonMask.THREE,
+              can_focus:   false,
+              label:       _("End session"),
+              reactive:    true,
+              style_class: 'button destroy-user-session',
+            };
+
+            this._killUserSessionButton = new St.Button(button_opts);
+            this._killUserSessionButton.connect('clicked',
+                    Lang.bind(this, function() {
+                      Util.spawn(['/usr/bin/gnome-session-quit',
+                                  '--force',
+                                  '--logout',
+                                  '--no-prompt']);
+                    }));
+            this.actor.add_child(this._killUserSessionButton);
+            this._killUserSessionButton.show();
+        }
+
         this._defaultButtonWell = new St.Widget({ layout_manager: new Clutter.BinLayout() });
         this._defaultButtonWellActor = null;
 
@@ -180,6 +295,7 @@
             if (!this._userVerifier.hasPendingMessages)
                 this._fadeOutMessage();
 
+            this._updateGuestButtonSensitivity(this._entry.text.length == 0 && !this._userNameEntered);
             this._updateNextButtonSensitivity(this._entry.text.length > 0 || this.verificationStatus == AuthPromptStatus.VERIFYING);
         });
         this._entry.clutter_text.connect('activate', () => {
@@ -402,6 +518,17 @@
         }
     },
 
+    _updateGuestButtonSensitivity(sensitive) {
+        if (this._guestuser_login !== 'guest') {
+            // forcibly hide the guest login button if guest logins are disabled
+            this.guestButton.hide();
+            return;
+        }
+
+        this.guestButton.reactive = sensitive;
+        this.guestButton.can_focus = sensitive;
+    },
+
     _updateNextButtonSensitivity(sensitive) {
         this.nextButton.reactive = sensitive;
         this.nextButton.can_focus = sensitive;
@@ -442,6 +569,18 @@
         this.nextButton.label = _("Next");
         this._preemptiveAnswer = null;
 
+        if (this._userName)
+            this._userName.text = " ";
+
+        this._userNameEntered = false;
+
+        // We're back at the start, so the guest button is usable again
+        // in case we are not reauthenticating.
+        if (this._mode != AuthPromptMode.UNLOCK_ONLY)
+            this.guestButton.show();
+        // this will also hide guest button if needed:
+        this._updateGuestButtonSensitivity(true);
+
         if (this._userVerifier)
             this._userVerifier.cancel();
 
diff -ruN gnome-shell-3.30.2.orig/js/gdm/loginDialog.js gnome-shell-3.30.2/js/gdm/loginDialog.js
--- gnome-shell-3.30.2.orig/js/gdm/loginDialog.js	2019-04-17 13:31:18.973752260 +0300
+++ gnome-shell-3.30.2/js/gdm/loginDialog.js	2019-04-17 13:32:07.150639982 +0300
@@ -1123,6 +1123,7 @@
     _showUserList() {
         this._ensureUserListLoaded();
         this._authPrompt.hide();
+        this._authPrompt._updateGuestButtonSensitivity(true);
         this._hideBannerView();
         this._sessionMenuButton.close();
         this._setUserListExpanded(true);
@@ -1132,6 +1133,7 @@
 
     _beginVerificationForItem(item) {
         this._authPrompt.setUser(item.user);
+        this._authPrompt.guestButton.hide();
 
         let userName = item.user.get_user_name();
         let hold = new Batch.Hold();
diff -ruN gnome-shell-3.30.2.orig/js/gdm/util.js gnome-shell-3.30.2/js/gdm/util.js
--- gnome-shell-3.30.2.orig/js/gdm/util.js	2019-04-17 13:31:18.973752260 +0300
+++ gnome-shell-3.30.2/js/gdm/util.js	2019-04-17 13:32:07.150639982 +0300
@@ -31,6 +31,7 @@
 var BANNER_MESSAGE_KEY = 'banner-message-enable';
 var BANNER_MESSAGE_TEXT_KEY = 'banner-message-text';
 var ALLOWED_FAILURES_KEY = 'allowed-failures';
+var GUESTUSER_KEY = 'enable-guestuser';
 
 var LOGO_KEY = 'logo';
 var DISABLE_USER_LIST_KEY = 'disable-user-list';
@@ -541,10 +542,9 @@
         // After that, we go back to the welcome screen.
 
         this._failCounter++;
-        let canRetry = retry && this._userName &&
-            (this._reauthOnly ||
-             this._failCounter < this._settings.get_int(ALLOWED_FAILURES_KEY));
-
+        let canRetry = retry && this._userName;     // let the user re-enter the password as
+                                                    // many times as they want, they aren't
+                                                    // going to brute-force it this way...
         if (canRetry) {
             if (!this.hasPendingMessages) {
                 this._retry();
diff -ruN gnome-shell-3.30.2.orig/po/de.po gnome-shell-3.30.2/po/de.po
--- gnome-shell-3.30.2.orig/po/de.po	2019-04-17 13:31:18.973752260 +0300
+++ gnome-shell-3.30.2/po/de.po	2019-04-17 13:32:07.150639982 +0300
@@ -352,6 +352,10 @@
 msgid "There was an error loading the preferences dialog for %s:"
 msgstr "Beim Laden des Einstellungsdialogs für %s ist ein Fehler aufgetreten:"
 
+#: js/gdm/authPrompt.js:141
+msgid "Login as guest"
+msgstr "Als Gast anmelden"
+
 #: js/gdm/authPrompt.js:147 js/ui/audioDeviceSelection.js:71
 #: js/ui/components/networkAgent.js:117 js/ui/components/polkitAgent.js:153
 #: js/ui/endSessionDialog.js:482 js/ui/extensionDownloader.js:197
@@ -373,6 +377,10 @@
 msgid "Sign In"
 msgstr "Anmelden"
 
+#: js/gdm/authPrompt.js:224
+msgid "End session"
+msgstr "Sitzung beenden"
+
 #: js/gdm/loginDialog.js:319
 msgid "Choose Session"
 msgstr "Sitzung wählen"
diff -ruN gnome-shell-3.30.2.orig/po/fi.po gnome-shell-3.30.2/po/fi.po
--- gnome-shell-3.30.2.orig/po/fi.po	2019-04-17 13:31:18.977752334 +0300
+++ gnome-shell-3.30.2/po/fi.po	2019-04-17 13:32:07.154640055 +0300
@@ -72,6 +72,10 @@
 msgid "Configure GNOME Shell Extensions"
 msgstr "Hallitse Gnome Shell -laajennuksia"
 
+#: js/gdm/authPrompt.js:141
+msgid "Login as guest"
+msgstr "Kirjaudu vieraana"
+
 #: data/org.gnome.Shell.desktop.in.in:4
 msgid "GNOME Shell"
 msgstr "Gnome Shell"
@@ -380,6 +384,10 @@
 msgid "Sign In"
 msgstr "Kirjaudu sisään"
 
+#: js/gdm/authPrompt.js:224
+msgid "End session"
+msgstr "Lopeta istunto"
+
 #: js/gdm/loginDialog.js:319
 msgid "Choose Session"
 msgstr "Valitse istunto"
diff -ruN gnome-shell-3.30.2.orig/po/fr.po gnome-shell-3.30.2/po/fr.po
--- gnome-shell-3.30.2.orig/po/fr.po	2019-04-17 13:31:18.977752334 +0300
+++ gnome-shell-3.30.2/po/fr.po	2019-04-17 13:32:07.154640055 +0300
@@ -64,6 +64,10 @@
 msgid "Configure GNOME Shell Extensions"
 msgstr "Configurer les extensions Shell de GNOME"
 
+#: js/gdm/authPrompt.js:141
+msgid "Login as guest"
+msgstr "Connectez-vous comme invité"
+
 #: data/org.gnome.Shell.desktop.in.in:4
 msgid "GNOME Shell"
 msgstr "Shell de GNOME"
@@ -382,6 +386,10 @@
 msgid "Sign In"
 msgstr "Se connecter"
 
+#: js/gdm/authPrompt.js:224
+msgid "End session"
+msgstr "Fin de session"
+
 #: js/gdm/loginDialog.js:319
 msgid "Choose Session"
 msgstr "Choisir une session"
diff -ruN gnome-shell-3.30.2.orig/po/sv.po gnome-shell-3.30.2/po/sv.po
--- gnome-shell-3.30.2.orig/po/sv.po	2019-04-17 13:31:18.977752334 +0300
+++ gnome-shell-3.30.2/po/sv.po	2019-04-17 13:32:07.154640055 +0300
@@ -331,6 +331,10 @@
 msgid "There was an error loading the preferences dialog for %s:"
 msgstr "Det uppstod ett fel vid inläsning av inställningsdialogen för %s:"
 
+#: js/gdm/authPrompt.js:141
+msgid "Login as guest"
+msgstr "Logga in som gäst"
+
 #: js/gdm/authPrompt.js:147 js/ui/audioDeviceSelection.js:71
 #: js/ui/components/networkAgent.js:117 js/ui/components/polkitAgent.js:153
 #: js/ui/endSessionDialog.js:482 js/ui/extensionDownloader.js:197
@@ -352,6 +356,10 @@
 msgid "Sign In"
 msgstr "Logga in"
 
+#: js/gdm/authPrompt.js:224
+msgid "End session"
+msgstr "Avsluta sessionen"
+
 #: js/gdm/loginDialog.js:319
 msgid "Choose Session"
 msgstr "Välj session"
