diff -ruN gnome-shell-3.30.2.build-5/js/ui/status/network.js gnome-shell-3.30.2.build-6/js/ui/status/network.js
--- gnome-shell-3.30.2.build-5/js/ui/status/network.js	2019-04-17 12:45:11.472940702 +0300
+++ gnome-shell-3.30.2.build-6/js/ui/status/network.js	2019-04-17 12:54:46.826158478 +0300
@@ -332,8 +332,7 @@
     },
 
     _autoConnect() {
-        let connection = new NM.SimpleConnection();
-        this._client.add_and_activate_connection_async(connection, this._device, null, null, null);
+        this._addAndActivateNewConnection(null);
     },
 
     destroy() {
@@ -923,14 +922,43 @@
                 Util.spawn(['gnome-control-center', 'wifi', 'connect-8021x-wifi',
                             this._device.get_path(), accessPoints[0].get_path()]);
             } else {
-                let connection = new NM.SimpleConnection();
-                this._client.add_and_activate_connection_async(connection, this._device, accessPoints[0].get_path(), null, null)
+                this._addAndActivateNewConnection(accessPoints[0].dbus_path);
             }
         }
 
         this.close();
     },
 
+    _addAndActivateNewConnection(dbus_path) {
+      let connection = new NetworkManager.Connection();
+
+      let cmd = '/usr/lib/puavo-ltsp-client/puavo-network-connections --ask-if-must-be-private';
+      let [res, out, err, status] = GLib.spawn_command_line_sync(cmd);
+
+      if (res && status == 0) {
+          // We do a similar trick as "nmcli"-utility does when given
+          // "private yes" as parameters.  Set the user of the network
+          // connection to the current user only so that network connection
+          // can be made with
+          // "org.freedesktop.NetworkManager.settings.modify.own" permission
+          // only and "org.freedesktop.NetworkManager.settings.modify.system"
+          // is not needed.  But this is intentionally coded so that if
+          // the above command is not found, the default behaviour of using
+          // system-level permission is retained.
+          let username = GLib.get_user_name();
+
+          if (username && username != "") {
+              let s_con = new NetworkManager.SettingConnection();
+              connection.add_setting(s_con);
+
+              s_con.add_permission("user", username, null);
+          }
+      }
+
+      this._client.add_and_activate_connection(connection, this._device,
+          dbus_path, null);
+    },
+
     _notifySsidCb(accessPoint) {
         if (accessPoint.get_ssid() != null) {
             accessPoint.disconnect(accessPoint._notifySsidId);
