diff -ruN gnome-shell-3.30.2.build-6/js/ui/screenShield.js gnome-shell-3.30.2.build-7/js/ui/screenShield.js
--- gnome-shell-3.30.2.build-6/js/ui/screenShield.js	2018-11-14 00:50:31.000000000 +0200
+++ gnome-shell-3.30.2.build-7/js/ui/screenShield.js	2019-04-17 12:57:56.389629168 +0300
@@ -726,6 +726,11 @@
     },
 
     _prepareForSleep(loginManager, aboutToSuspend) {
+        // do not activate screenShield on greeter when going to suspend
+        if (Main.sessionMode.isGreeter) {
+            return;
+        }
+
         if (aboutToSuspend) {
             if (this._settings.get_boolean(LOCK_ENABLED_KEY))
                 this.lock(true);
