diff -ruN gnome-shell/js/gdm/loginDialog.js gnome-shell.2/js/gdm/loginDialog.js
--- gnome-shell/js/gdm/loginDialog.js	2023-02-14 02:11:32.000000000 +0200
+++ gnome-shell/js/gdm/loginDialog.js	2023-03-16 12:41:31.412029059 +0200
@@ -887,6 +887,8 @@
         } else {
             this._hideUserListAndBeginVerification();
         }
+
+        GLib.spawn_command_line_async('/usr/bin/puavo-autopilot-login');
     }
 
     _onDefaultSessionChanged(client, sessionId) {
