diff -ruN gnome-shell/js/ui/messageTray.js gnome-shell.5/js/ui/messageTray.js
--- gnome-shell/js/ui/messageTray.js	2023-02-14 02:11:32.000000000 +0200
+++ gnome-shell/js/ui/messageTray.js	2023-03-16 12:53:26.571191327 +0200
@@ -1265,11 +1265,9 @@
         this._notification.acknowledged = true;
         this._notification.playSound();
 
-        // We auto-expand notifications with CRITICAL urgency, or for which the relevant setting
-        // is on in the control center.
-        if (this._notification.urgency == Urgency.CRITICAL ||
-            this._notification.source.policy.forceExpanded)
-            this._expandBanner(true);
+        // Always auto-expand all notifications. Many users just don't understand that notifications
+        // can be expanded and that they can contain buttons. So make everything visible, always.
+        this._expandBanner(true);
 
         // We tween all notifications to full opacity. This ensures that both new notifications and
         // notifications that might have been in the process of hiding get full opacity.
