diff -ruN gnome-shell/js/gdm/loginDialog.js gnome-shell/js/gdm/loginDialog.js
--- gnome-shell/js/gdm/loginDialog.js	2025-05-14 12:37:44.543291309 +0300
+++ gnome-shell/js/gdm/loginDialog.js	2025-05-14 13:02:13.437505908 +0300
@@ -42,6 +42,7 @@
 import * as UserWidget from '../ui/userWidget.js';
 
 const _FADE_ANIMATION_TIME = 250;
+const _LOGO_ICON_HEIGHT = 48;
 const _SCROLL_ANIMATION_TIME = 500;
 const _TIMED_LOGIN_IDLE_THRESHOLD = 5.0;
 const _CONFLICTING_SESSION_DIALOG_TIMEOUT = 60;
@@ -969,7 +970,7 @@
             const scaleFactor = St.ThemeContext.get_for_stage(global.stage).scale_factor;
             const texture = this._textureCache.load_file_async(
                 this._logoFile,
-                -1, -1,
+                -1, _LOGO_ICON_HEIGHT,
                 scaleFactor, resourceScale);
             this._logoBin.add_child(texture);
         }
