diff -ruN gnome-shell/js/gdm/authPrompt.js gnome-shell.8/js/gdm/authPrompt.js
--- gnome-shell/js/gdm/authPrompt.js	2023-02-14 02:11:32.000000000 +0200
+++ gnome-shell/js/gdm/authPrompt.js	2023-03-16 12:59:31.808819458 +0200
@@ -188,6 +188,8 @@
         ShellEntry.addContextMenu(this._textEntry, { actionMode: Shell.ActionMode.NONE });
 
         this._passwordEntry = new St.PasswordEntry(entryParams);
+        // Do not allow password peeking in Puavo OS.
+        this._passwordEntry.set_show_peek_icon(false);
         ShellEntry.addContextMenu(this._passwordEntry, { actionMode: Shell.ActionMode.NONE });
 
         this._entry = this._passwordEntry;
diff -ruN gnome-shell/js/ui/shellEntry.js gnome-shell.8/js/ui/shellEntry.js
--- gnome-shell/js/ui/shellEntry.js	2023-02-14 02:11:32.000000000 +0200
+++ gnome-shell/js/ui/shellEntry.js	2023-03-16 12:59:31.808819458 +0200
@@ -36,8 +36,14 @@
 
     _makePasswordItem() {
         let item = new PopupMenu.PopupMenuItem('');
-        item.connect('activate', this._onPasswordActivated.bind(this));
-        this.addMenuItem(item);
+        // Do not add this item to menu or active it ever!
+        // This is a design mistake, user passwords end up showing accidentally
+        // sometimes.  For example, if user writes username+password and
+        // presses enter three times after that in a fast succession, the
+        // password will appear on the screen before the login proceeds
+        // to the desktop.
+        // item.connect('activate', this._onPasswordActivated.bind(this));
+        // this.addMenuItem(item);
         this._passwordItem = item;
 
         this._entry.bind_property('show-peek-icon',
@@ -96,7 +102,9 @@
     }
 
     _onPasswordActivated() {
-        this._entry.password_visible  = !this._entry.password_visible;
+        // Password should never be visible in Puavo OS.
+        // this._entry.password_visible  = !this._entry.password_visible;
+        this._entry.password_visible = false;
     }
 };
 
