diff -ruN gnome-shell/js/ui/main.js gnome-shell/js/ui/main.js
--- gnome-shell/js/ui/main.js	2026-04-10 15:55:31.794556542 +0300
+++ gnome-shell/js/ui/main.js	2026-04-10 15:48:06.578123070 +0300
@@ -371,10 +371,6 @@
             }
         }
 
-        if (sessionMode.currentMode !== 'gdm' &&
-            sessionMode.currentMode !== 'initial-setup')
-            _handleLockScreenWarning();
-
         LoginManager.registerSessionWithGDM();
 
         if (perfModule) {
@@ -392,32 +388,6 @@
     }
 }
 
-async function _handleLockScreenWarning() {
-    const path = `${global.userdatadir}/lock-warning-shown`;
-    const file = Gio.File.new_for_path(path);
-
-    const hasLockScreen = screenShield !== null;
-    if (hasLockScreen) {
-        try {
-            await file.delete_async(0, null);
-        } catch (e) {
-            if (!e.matches(Gio.IOErrorEnum, Gio.IOErrorEnum.NOT_FOUND))
-                logError(e);
-        }
-    } else {
-        try {
-            if (!await file.touch_async())
-                return;
-        } catch (e) {
-            logError(e);
-        }
-
-        notify(
-            _('Screen Lock disabled'),
-            _('Screen Locking requires the GNOME display manager'));
-    }
-}
-
 function _getStylesheet(name) {
     let stylesheet;
 
