diff -ruN gnome-shell/js/gdm/popupMenu.js gnome-shell/js/gdm/popupMenu.js
--- gnome-shell/js/ui/popupMenu.js	2023-09-16 12:57:40.000000000 +0300
+++ gnome-shell/js/ui/popupMenu.js	2025-10-13 13:57:43.337825727 +0300
@@ -3,7 +3,7 @@
             PopupImageMenuItem, PopupMenu, PopupDummyMenu, PopupSubMenu,
             PopupMenuSection, PopupSubMenuMenuItem, PopupMenuManager */
 
-const { Atk, Clutter, Gio, GObject, Graphene, Shell, St } = imports.gi;
+const { Atk, Clutter, Gio, GLib, GObject, Graphene, Shell, St } = imports.gi;
 const Signals = imports.misc.signals;
 
 const BoxPointer = imports.ui.boxpointer;
@@ -563,6 +563,12 @@
     }
 
     addSettingsAction(title, desktopFile) {
+        let puavo_exammode_session = GLib.getenv('EXAMMODE_SESSION');
+        if (puavo_exammode_session && puavo_exammode_session !== '') {
+            // do not allow settings actions in Puavo examination mode
+            return new PopupMenuItem(title);
+        }
+
         let menuItem = this.addAction(title, () => {
             let app = Shell.AppSystem.get_default().lookup_app(desktopFile);
 
