--- a/src/java.base/share/classes/java/lang/Float.java	2017-11-16 00:58:50.000000000 +0100
+++ b/src/java.base/share/classes/java/lang/Float.java	2017-12-22 22:56:30.771249660 +0100
@@ -84,7 +84,9 @@
      *
      * @since 1.6
      */
-    public static final float MIN_NORMAL = 0x1.0p-126f; // 1.17549435E-38f
+     // FIXME: still required on alpha?
+     // public static final float MIN_NORMAL = 0x1.0p-126f; // 1.17549435E-38f
+     public static final float MIN_NORMAL = Float.intBitsToFloat(0x00800000);
 
     /**
      * A constant holding the smallest positive nonzero value of type
