--- a/grub-core/disk/loopback.c	2024-06-07 12:35:49.378071026 +0300
+++ b/grub-core/disk/loopback.c	2024-06-07 12:37:19.825065864 +0300
@@ -98,7 +98,8 @@
       return grub_error (GRUB_ERR_BAD_ARGUMENT, "device name already exists");
 
   file = grub_file_open (args[1], GRUB_FILE_TYPE_LOOPBACK
-			 | GRUB_FILE_TYPE_NO_DECOMPRESS);
+			 | GRUB_FILE_TYPE_NO_DECOMPRESS
+			 | GRUB_FILE_TYPE_SKIP_SIGNATURE);
   if (! file)
     return grub_errno;
 
