Description: Make the build reproducible
Author: Chris Lamb <lamby@debian.org>
Last-Update: 2018-04-11

--- libmypaint-1.3.0.orig/doc/Doxyfile
+++ libmypaint-1.3.0/doc/Doxyfile
@@ -119,7 +119,7 @@ INLINE_INHERITED_MEMB  = NO
 # path before files name in the file list and in the header files. If set
 # to NO the shortest path that makes the file name unique will be used.
 
-FULL_PATH_NAMES        = YES
+FULL_PATH_NAMES        = NO
 
 # If the FULL_PATH_NAMES tag is set to YES then the STRIP_FROM_PATH tag
 # can be used to strip a user-defined part of the path. Stripping is
