description: set default printer resolution to 600dpi
author: Ralf E-Mail <anonymous272912@gmail.com>
bug-debian: https://bugs.debian.org/687168

--- a/dlls/winspool.drv/generic.ppd
+++ b/dlls/winspool.drv/generic.ppd
@@ -116,7 +116,7 @@
 
 *OpenUI *Resolution/Output Resolution: PickOne
 *OrderDependency: 20 AnySetup *Resolution
-*DefaultResolution: 300dpi
+*DefaultResolution: 600dpi
 *Resolution 150dpi/150 DPI: "<</HWResolution[150 150]>>setpagedevice"
 *Resolution 300dpi/300 DPI: "<</HWResolution[300 300]>>setpagedevice"
 *Resolution 600dpi/600 DPI: "<</HWResolution[600 600]>>setpagedevice"
