From: Andrew Chadwick <a.t.chadwick@gmail.com>
Date: Sun, 8 Jan 2017 17:14:18 +0000
Subject: Fix cherry-pick: stable branch has pygtk syntax

Origin: upstream, 1.2.1
Bug-Debian: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=859064
---
 gui/mode.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/gui/mode.py b/gui/mode.py
index 2802720..c99eb3d 100644
--- a/gui/mode.py
+++ b/gui/mode.py
@@ -811,7 +811,7 @@ class SingleClickMode (InteractionMode):
 
     #: The cursor to use when entering the mode
     # FIXME: Use Gdk.Cursor.new_for_display; read-only property
-    cursor = Gdk.Cursor.new(Gdk.CursorType.ARROW)
+    cursor = gdk.Cursor.new(gdk.CursorType.ARROW)
 
     def __init__(self, ignore_modifiers=False, **kwds):
         super(SingleClickMode, self).__init__(**kwds)
@@ -860,7 +860,7 @@ class DragMode (InteractionMode):
     """
 
     # FIXME: Use Gdk.Cursor.new_for_display; read-only property
-    inactive_cursor = Gdk.Cursor.new(Gdk.CursorType.ARROW)
+    inactive_cursor = gdk.Cursor.new(gdk.CursorType.ARROW)
     active_cursor = None
 
     #: If true, exit mode when initial modifiers are released
