--- veyon-4.7.3+repack1.orig/plugins/platform/linux/LinuxInputDeviceFunctions.cpp
+++ veyon-4.7.3+repack1/plugins/platform/linux/LinuxInputDeviceFunctions.cpp
@@ -33,8 +33,6 @@ void LinuxInputDeviceFunctions::enableIn
 {
 	if( m_inputDevicesDisabled )
 	{
-		restoreKeyMapTable();
-
 		m_inputDevicesDisabled = false;
 	}
 }
@@ -45,8 +43,6 @@ void LinuxInputDeviceFunctions::disableI
 {
 	if( m_inputDevicesDisabled == false )
 	{
-		setEmptyKeyMapTable();
-
 		m_inputDevicesDisabled = true;
 	}
 }
