diff -ruN gnome-network-displays.orig/src/nd-wfd-p2p-sink.c gnome-network-displays/src/nd-wfd-p2p-sink.c
--- gnome-network-displays.orig/src/nd-wfd-p2p-sink.c   2023-05-16 11:56:44.393996265 +0300
+++ gnome-network-displays/src/nd-wfd-p2p-sink.c        2023-05-16 11:59:12.454782909 +0300
@@ -472,6 +472,7 @@
   connection = nm_simple_connection_new ();
 
   general_setting = nm_setting_connection_new ();
+  nm_setting_connection_add_permission(general_setting, "user", g_get_user_name(), NULL);
   nm_connection_add_setting (connection, general_setting);
   g_object_set (general_setting, NM_SETTING_CONNECTION_ZONE, ND_WFD_ZONE, NULL);

