description: fix uninitialized variable warnings on arm64
author: Michael Gilbert <mgilbert@debian.org>

--- a/dlls/devenum/createdevenum.c
+++ b/dlls/devenum/createdevenum.c
@@ -388,7 +388,7 @@ static void write_filter_data(IPropertyB
     HRESULT hr;
 
     hr = CoCreateInstance(&CLSID_FilterMapper2, NULL, CLSCTX_INPROC, &IID_IAMFilterData, (void **)&fildata);
-    if (FAILED(hr)) goto cleanup;
+    if (FAILED(hr)) goto cleanup2;
 
     hr = IAMFilterData_CreateFilterData(fildata, rgf, &data, &size);
     if (FAILED(hr)) goto cleanup;
@@ -409,8 +409,9 @@ static void write_filter_data(IPropertyB
     if (FAILED(hr)) goto cleanup;
 
 cleanup:
-    VariantClear(&var);
     CoTaskMemFree(data);
+cleanup2:
+    VariantClear(&var);
     IAMFilterData_Release(fildata);
 }
 
--- a/dlls/devenum/mediacatenum.c
+++ b/dlls/devenum/mediacatenum.c
@@ -128,7 +128,7 @@ static HRESULT WINAPI DEVENUM_IPropertyB
     HRESULT res = S_OK;
     LONG reswin32 = ERROR_SUCCESS;
     WCHAR name[80];
-    HKEY hkey;
+    HKEY hkey = NULL;
 
     TRACE("(%p)->(%s, %p, %p)\n", This, debugstr_w(pszPropName), pVar, pErrorLog);
 
--- a/dlls/user32/sysparams.c
+++ b/dlls/user32/sysparams.c
@@ -1733,7 +1733,7 @@ BOOL WINAPI SystemParametersInfoW( UINT
     case SPI_GETNONCLIENTMETRICS:
     {
         LPNONCLIENTMETRICSW lpnm = pvParam;
-        int padded_border;
+        int padded_border = 0;
 
         if (!pvParam) return FALSE;
 
