description: disable functions that are unused on arm64
author: Michael Gilbert <mgilbert@debian.org>

--- a/dlls/dmusic/collection.c
+++ b/dlls/dmusic/collection.c
@@ -49,10 +49,12 @@ static inline IDirectMusicCollectionImpl
     return CONTAINING_RECORD(iface, IDirectMusicCollectionImpl, IDirectMusicCollection_iface);
 }
 
+#ifndef __aarch64__
 static inline struct dmobject *impl_from_IDirectMusicObject(IDirectMusicObject *iface)
 {
     return CONTAINING_RECORD(iface, struct dmobject, IDirectMusicObject_iface);
 }
+#endif
 
 static inline IDirectMusicCollectionImpl *impl_from_IPersistStream(IPersistStream *iface)
 {
--- a/dlls/dwrite/font.c
+++ b/dlls/dwrite/font.c
@@ -5103,10 +5103,12 @@ static HRESULT WINAPI glyphrunanalysis_G
     return S_OK;
 }
 
+#ifndef __aarch64__
 static inline int get_dib_stride( int width, int bpp )
 {
     return ((width * bpp + 31) >> 3) & ~3;
 }
+#endif
 
 static inline BYTE *get_pixel_ptr(BYTE *ptr, DWRITE_TEXTURE_TYPE type, const RECT *runbounds, const RECT *bounds)
 {
--- a/dlls/windowscodecs/converter.c
+++ b/dlls/windowscodecs/converter.c
@@ -80,12 +80,14 @@ typedef struct FormatConverter {
     CRITICAL_SECTION lock; /* must be held when initialized */
 } FormatConverter;
 
+#ifndef __aarch64__
 /* https://www.w3.org/Graphics/Color/srgb */
 static inline float from_sRGB_component(float f)
 {
     if (f <= 0.04045f) return f / 12.92f;
     return powf((f + 0.055f) / 1.055f, 2.4f);
 }
+#endif
 
 static inline float to_sRGB_component(float f)
 {
--- a/dlls/dmscript/script.c
+++ b/dlls/dmscript/script.c
@@ -203,11 +203,13 @@ static const IDirectMusicScriptVtbl dmsc
     IDirectMusicScriptImpl_EnumVariable
 };
 
+#ifndef __aarch64__
 /* IDirectMusicScriptImpl IDirectMusicObject part: */
 static inline IDirectMusicScriptImpl *impl_from_IDirectMusicObject(IDirectMusicObject *iface)
 {
     return CONTAINING_RECORD(iface, IDirectMusicScriptImpl, dmobj.IDirectMusicObject_iface);
 }
+#endif
 
 static HRESULT WINAPI script_IDirectMusicObject_ParseDescriptor(IDirectMusicObject *iface,
         IStream *stream, DMUS_OBJECTDESC *desc)
--- a/dlls/d2d1/device.c
+++ b/dlls/d2d1/device.c
@@ -210,11 +210,6 @@ static inline struct d2d_device_context
     return CONTAINING_RECORD(iface, struct d2d_device_context, ID2D1DeviceContext_iface);
 }
 
-static inline struct d2d_device_context *impl_from_ID2D1RenderTarget(ID2D1RenderTarget *iface)
-{
-    return CONTAINING_RECORD(iface, struct d2d_device_context, ID2D1DeviceContext_iface);
-}
-
 static HRESULT STDMETHODCALLTYPE d2d_device_context_inner_QueryInterface(IUnknown *iface, REFIID iid, void **out)
 {
     struct d2d_device_context *context = impl_from_IUnknown(iface);
