From: Didier Raboud <odyx@debian.org>
Date: Tue, 9 Aug 2016 18:11:24 +0200
Subject: Debian: Move cupsd.conf.default from SERVERROOT to DATADIR

Origin: vendor
Bug: https://github.com/apple/cups/issues/4342
Bug: https://github.com/OpenPrinting/cups/pull/26
Bug-Debian: https://bugs.debian.org/640124
---
 cgi-bin/admin.c | 6 +++++-
 conf/Makefile   | 5 +++--
 2 files changed, 8 insertions(+), 3 deletions(-)

diff --git a/cgi-bin/admin.c b/cgi-bin/admin.c
index 37f93b5..ac9cc42 100644
--- a/cgi-bin/admin.c
+++ b/cgi-bin/admin.c
@@ -1635,6 +1635,7 @@ do_config_server(http_t *http)		/* I - HTTP connection */
     int		ch;			/* Character from file */
     char	filename[1024];		/* Filename */
     const char	*server_root;		/* Location of config files */
+    const char	*data_dir;		/* Location of data files */
 
 
    /*
@@ -1718,7 +1719,10 @@ do_config_server(http_t *http)		/* I - HTTP connection */
     * well...
     */
 
-    strlcat(filename, ".default", sizeof(filename));
+    if ((data_dir = getenv("CUPS_DATADIR")) == NULL)
+      data_dir = CUPS_DATADIR;
+
+    snprintf(filename, sizeof(filename), "%s/cupsd.conf.default",data_dir);
 
     if (!stat(filename, &info) && info.st_size < (1024 * 1024) &&
         (cupsd = cupsFileOpen(filename, "r")) != NULL)
diff --git a/conf/Makefile b/conf/Makefile
index e249e60..42a123f 100644
--- a/conf/Makefile
+++ b/conf/Makefile
@@ -70,7 +70,7 @@ install-data:
 		else \
 			$(INSTALL_CONFIG) -g $(CUPS_GROUP) $$file $(SERVERROOT) ; \
 		fi ; \
-		$(INSTALL_CONFIG) -g $(CUPS_GROUP) $$file $(SERVERROOT)/$$file.default; \
+		$(INSTALL_CONFIG) -g $(CUPS_GROUP) $$file $(DATADIR)/$$file.default; \
 	done
 	$(INSTALL_DIR) -m 755 $(DATADIR)/mime
 	for file in $(REPLACE); do \
@@ -118,9 +118,10 @@ install-libs:
 #
 
 uninstall:
-	for file in $(KEEP) $(REPLACE) cupsd.conf.default; do \
+	for file in $(KEEP) $(REPLACE); do \
 		$(RM) $(SERVERROOT)/$$file; \
 	done
+	$(RM) $(DATADIR)/cupsd.conf.default
 	-$(RMDIR) $(SERVERROOT)
 	for file in $(REPLACE); do \
 		$(RM) $(DATADIR)/mime/$$file; \
