From 545d46fb0bf1cd8414ab28148f3a3126c3cf75fe Mon Sep 17 00:00:00 2001
From: Michael R Sweet <michael.r.sweet@gmail.com>
Date: Thu, 1 Aug 2019 13:02:35 -0400
Subject: Fix memory leak in ppdOpen

Closes: #946941
---
 cups/ppd.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/cups/ppd.c b/cups/ppd.c
index 435b992f4..b823d17b4 100644
--- a/cups/ppd.c
+++ b/cups/ppd.c
@@ -716,6 +716,8 @@ _ppdOpen(
 	   strncmp(ll, keyword, ll_len)))
       {
 	DEBUG_printf(("2_ppdOpen: Ignoring localization: \"%s\"\n", keyword));
+	free(string);
+	string = NULL;
 	continue;
       }
       else if (localization == _PPD_LOCALIZATION_ICC_PROFILES)
@@ -735,6 +737,8 @@ _ppdOpen(
 	if (i >= (int)(sizeof(color_keywords) / sizeof(color_keywords[0])))
 	{
 	  DEBUG_printf(("2_ppdOpen: Ignoring localization: \"%s\"\n", keyword));
+	  free(string);
+	  string = NULL;
 	  continue;
 	}
       }
