diff -ruN cups.orig/cups/dest.c cups/cups/dest.c
--- cups.orig/cups/dest.c	2021-02-01 23:10:25.000000000 +0200
+++ cups/cups/dest.c	2021-09-06 13:04:30.950902232 +0300
@@ -1648,7 +1648,7 @@
               cups_dest_t **dests)	/* O - Destinations */
 {
   _cups_getdata_t data;                 /* Enumeration data */
-
+  char *show_avahi_printers;
 
   DEBUG_printf(("cupsGetDests2(http=%p, dests=%p)", (void *)http, (void *)dests));
 
@@ -1684,7 +1684,8 @@
   data.num_dests = 0;
   data.dests     = NULL;
 
-  if (!httpAddrLocalhost(httpGetAddress(http)))
+  show_avahi_printers = getenv("CUPS_SHOW_AVAHI_PRINTERS");
+  if ((show_avahi_printers != NULL && strcmp(show_avahi_printers, "false") == 0) || !httpAddrLocalhost(httpGetAddress(http)))
   {
    /*
     * When talking to a remote cupsd, just enumerate printers on the remote
