diff -ruN mutter.0/src/backends/meta-monitor-config-migration.c mutter.5/src/backends/meta-monitor-config-migration.c
--- mutter.0/src/backends/meta-monitor-config-migration.c	2018-07-14 21:49:25.000000000 +0300
+++ mutter.5/src/backends/meta-monitor-config-migration.c	2019-04-19 09:57:24.932363764 +0300
@@ -37,6 +37,7 @@
 #include "backends/meta-monitor-config-migration.h"
 
 #include <gio/gio.h>
+#include <stdlib.h>
 #include <string.h>
 
 #include "backends/meta-monitor-config-manager.h"
@@ -1158,9 +1159,14 @@
   g_autoptr (GFile) backup_file = NULL;
   g_autofree char *user_file_path = NULL;
   g_autoptr (GFile) user_file = NULL;
+  char *monitors_file_path;
+
+  monitors_file_path = getenv("MUTTER_MONITOR_FILENAME");
+  if (monitors_file_path == NULL)
+    monitors_file_path = "monitors.xml";
 
   user_file_path = g_build_filename (g_get_user_config_dir (),
-                                     "monitors.xml",
+                                     monitors_file_path,
                                      NULL);
   user_file = g_file_new_for_path (user_file_path);
   backup_path = g_build_filename (g_get_user_config_dir (),
@@ -1173,8 +1179,8 @@
                     NULL, NULL, NULL,
                     error))
     {
-      g_warning ("Failed to make a backup of monitors.xml: %s",
-                 (*error)->message);
+      g_warning ("Failed to make a backup of %s: %s",
+                 monitors_file_path, (*error)->message);
       g_clear_error (error);
     }
 
