description: rebuild automatically generated opengl files
author: Michael Gilbert <mgilbert@debian.org>

--- a/dlls/opengl32/make_opengl
+++ b/dlls/opengl32/make_opengl
@@ -315,13 +315,6 @@ if ($version eq "1.0") {
 }
 
 #
-# Fetch the registry files
-#
--f "gl.xml" || system "wget https://raw.github.com/KhronosGroup/OpenGL-Registry/master/xml/gl.xml" || die "cannot download gl.xml";
--f "wgl.xml" || system "wget https://raw.github.com/KhronosGroup/OpenGL-Registry/master/xml/wgl.xml" || die "cannot download wgl.xml";
-
-
-#
 # Then, create the list of all OpenGL functions using the registry
 # files. This will create two hash-tables, one with all the function
 # whose category matches the one listed in '@norm_categories', the other
@@ -559,21 +552,22 @@ sub parse_file($)
     }
 }
 
-parse_file( "gl.xml" );
-parse_file( "wgl.xml" );
+parse_file( "/usr/share/khronos-api/gl.xml" );
+parse_file( "/usr/share/khronos-api/wgl.xml" );
 parse_file( "winegl.xml" );
 
 #
 # Get the current wgl_driver.h version
+# Modified by -1, see debian/scripts/import
 #
-my $wgl_version = 0;
-open HEADER, "<$wgl_driver_file" or die "cannot open $wgl_driver_file";
-while (<HEADER>)
-{
-    next unless /^#define WINE_WGL_DRIVER_VERSION (\d+)/;
-    $wgl_version = $1;
-    last;
-}
+my $wgl_version = 18;
+
+#
+# Generate the wglext.h file
+#
+my $wglext_file = "../../include/wine/wglext.h";
+open HEADER, ">$wglext_file" or die "cannot create $wglext_file";
+print HEADER "#include <khronos-api/GL/wglext.h>";
 close HEADER;
 
 #
