description: use system unicode
author: Michael Gilbert <mgilbert@debian.org>

--- a/tools/make_unicode
+++ b/tools/make_unicode
@@ -21,12 +21,11 @@
 
 use strict;
 
-# base URLs for www.unicode.org files
+# Locations of unicode files
 my $MAPPINGS = "http://www.unicode.org/Public/MAPPINGS";
-my $UNIDATA = "http://www.unicode.org/Public/9.0.0/ucd";
-my $REPORTS = "http://www.unicode.org/reports";
-my $RFCS = "http://www.rfc-editor.org/rfc";
-my $VERTICALDATA = "http://www.unicode.org/Public/vertical/revision-15";
+my $UNIDATA = "/usr/share/unicode";
+my $REPORTS = "debian/unicode";
+my $RFCS = "debian";
 
 # Sort keys file
 my $SORTKEYS = "tr10/allkeys.txt";
@@ -191,7 +190,8 @@ my %matra_types =
     "Bottom_And_Right"  => 0x0b,
     "Top_And_Bottom_And_Right"  => 0x0c,
     "Overstruck"  => 0x0d,
-    "Invisible"  => 0x0e
+    "Invisible"  => 0x0e,
+    "Bottom_And_Left"  => 0x0f
 );
 
 my %nameprep_flags =
@@ -375,15 +375,8 @@ my %joining_forms =
 sub open_data_file($$)
 {
     my ($base, $name) = @_;
-    (my $dir = "data/$name") =~ s/\/[^\/]+$//;
     local *FILE;
-    unless (-f "data/$name")
-    {
-        print "Fetching $base/$name...\n";
-        system "mkdir", "-p", $dir;
-        !system "wget", "-q", "-O", "data/$name", "$base/$name" or die "cannot fetch $base/$name";
-    }
-    open FILE, "<data/$name" or die "cannot open data/$name";
+    open FILE, "<$base/$name" or die "cannot open $base/$name";
     return *FILE;
 }
 
@@ -1672,7 +1665,7 @@ sub dump_vertical($)
     my $filename = shift;
     my @vertical_table;
 
-    my $INPUT = open_data_file( $VERTICALDATA, "VerticalOrientation-15.txt" );
+    my $INPUT = open_data_file( $UNIDATA, "VerticalOrientation.txt" );
     while (<$INPUT>)
     {
         next if /^\#/;  # skip comments
@@ -1705,7 +1698,7 @@ sub dump_vertical($)
     open OUTPUT,">$filename.new" or die "Cannot create $filename";
     print "Building $filename\n";
     print OUTPUT "/* Unicode Vertical Orientation */\n";
-    print OUTPUT "/* generated from $VERTICALDATA/VerticalOrientation-15.txt */\n";
+    print OUTPUT "/* generated from $UNIDATA/VerticalOrientation.txt */\n";
     print OUTPUT "/* DO NOT EDIT!! */\n\n";
     print OUTPUT "#include \"wine/unicode.h\"\n\n";
 
@@ -2612,9 +2605,10 @@ dump_vertical( "dlls/gdi32/vertical.c" )
 dump_vertical( "dlls/wineps.drv/vertical.c" );
 dump_nameprep( "dlls/kernel32/nameprep.c" );
 
-foreach my $file (@allfiles) { HANDLE_FILE( @{$file} ); }
+# Skipping: unicode-data currently has no MAPPING files
+# foreach my $file (@allfiles) { HANDLE_FILE( @{$file} ); }
 
-output_cptable("libs/port/cptable.c");
+output_cptable("libs/port/cptable.generated");
 
 exit 0;
 
--- a/libs/port/cptable.c
+++ b/libs/port/cptable.c
@@ -22,8 +22,9 @@
 
 #include "wine/unicode.h"
 
-/* Everything below this line is generated automatically by make_unicode */
-/* ### cpmap begin ### */
+#include "cptable.generated"
+
+/*
 extern union cptable cptable_037;
 extern union cptable cptable_424;
 extern union cptable cptable_437;
@@ -172,8 +173,7 @@ static const union cptable * const cptab
     &cptable_28605,
     &cptable_28606,
 };
-/* ### cpmap end ### */
-/* Everything above this line is generated automatically by make_unicode */
+*/
 
 #define NB_CODEPAGES  (sizeof(cptables)/sizeof(cptables[0]))
 
