description: use system unicode
author: Michael Gilbert <mgilbert@debian.org>

--- a/tools/make_unicode
+++ b/tools/make_unicode
@@ -21,11 +21,11 @@
 
 use strict;
 
-# base URLs for www.unicode.org files
+# Locations of unicode files
 my $MAPPINGS = "http://www.unicode.org/Public/MAPPINGS";
-my $UNIDATA = "http://www.unicode.org/Public/11.0.0/ucd";
-my $REPORTS = "http://www.unicode.org/reports";
-my $RFCS = "http://www.rfc-editor.org/rfc";
+my $UNIDATA = "/usr/share/unicode";
+my $REPORTS = "debian/unicode";
+my $RFCS = "debian";
 
 # Sort keys file
 my $SORTKEYS = "tr10/allkeys.txt";
@@ -376,15 +376,8 @@ my %joining_forms =
 sub open_data_file($$)
 {
     my ($base, $name) = @_;
-    (my $dir = "data/$name") =~ s/\/[^\/]+$//;
     local *FILE;
-    unless (-f "data/$name")
-    {
-        print "Fetching $base/$name...\n";
-        system "mkdir", "-p", $dir;
-        !system "wget", "-q", "-O", "data/$name", "$base/$name" or die "cannot fetch $base/$name";
-    }
-    open FILE, "<data/$name" or die "cannot open data/$name";
+    open FILE, "<$base/$name" or die "cannot open $base/$name";
     return *FILE;
 }
 
@@ -2618,9 +2611,10 @@ dump_vertical( "dlls/gdi32/vertical.c" )
 dump_vertical( "dlls/wineps.drv/vertical.c" );
 dump_nameprep( "dlls/kernel32/nameprep.c" );
 
-foreach my $file (@allfiles) { HANDLE_FILE( @{$file} ); }
+# Skipping: unicode-data currently has no MAPPING files
+# foreach my $file (@allfiles) { HANDLE_FILE( @{$file} ); }
 
-output_cptable("libs/port/cptable.c");
+output_cptable("libs/port/cptable.generated");
 
 exit 0;
 
--- a/libs/port/cptable.c
+++ b/libs/port/cptable.c
@@ -22,8 +22,9 @@
 
 #include "wine/unicode.h"
 
-/* Everything below this line is generated automatically by make_unicode */
-/* ### cpmap begin ### */
+#include "cptable.generated"
+
+/*
 extern union cptable cptable_037;
 extern union cptable cptable_424;
 extern union cptable cptable_437;
@@ -172,8 +173,7 @@ static const union cptable * const cptab
     &cptable_28605,
     &cptable_28606,
 };
-/* ### cpmap end ### */
-/* Everything above this line is generated automatically by make_unicode */
+*/
 
 
 static int cmp_codepage( const void *codepage, const void *entry )
