diff -ruN cups-filters-1.28.7/utils/cups-browsed.c cups-filters-1.28.7.build.1/utils/cups-browsed.c
--- cups-filters-1.28.7/utils/cups-browsed.c	2021-01-07 20:44:26.000000000 +0200
+++ cups-filters-1.28.7/utils/cups-browsed.c	2022-06-29 10:34:42.505458406 +0300
@@ -5297,38 +5297,6 @@
 }
 
 int
-disable_printer (const char *printer, const char *reason) {
-  ipp_t *request;
-  char uri[HTTP_MAX_URI];
-  http_t *conn = NULL;
-
-  conn = http_connect_local ();
-  if (conn == NULL) {
-    debug_printf("Cannot connect to local CUPS to disable printer %s.\n",
-		 printer);
-    return -1;
-  }
-
-  if (reason == NULL)
-    reason = "Disabled by cups-browsed";
-  httpAssembleURIf(HTTP_URI_CODING_ALL, uri, sizeof(uri), "ipp", NULL,
-		   "localhost", 0, "/printers/%s", printer);
-  request = ippNewRequest (IPP_PAUSE_PRINTER);
-  ippAddString (request, IPP_TAG_OPERATION, IPP_TAG_URI,
-		"printer-uri", NULL, uri);
-  ippAddString (request, IPP_TAG_OPERATION, IPP_TAG_TEXT,
-		"printer-state-message", NULL, reason);
-  ippDelete(cupsDoRequest (conn, request, "/admin/"));
-  if (cupsLastError() > IPP_STATUS_OK_EVENTS_COMPLETE) {
-    debug_printf("ERROR: Failed disabling printer '%s': %s\n",
-		 printer, cupsLastErrorString());
-    return -1;
-  }
-  debug_printf("Disabled printer '%s'\n", printer);
-  return 0;
-}
-
-int
 set_cups_default_printer(const char *printer) {
   ipp_t *request;
   char uri[HTTP_MAX_URI];
@@ -7815,15 +7783,6 @@
 	  if (num_jobs > 0) { /* There are still jobs */
 	    debug_printf("Queue has still jobs or CUPS error!\n");
 	    cupsFreeJobs(num_jobs, jobs);
-	    /* Disable the queue */
-#ifdef HAVE_AVAHI
-	    if (avahi_present || p->domain == NULL || p->domain[0] == '\0')
-	      /* If avahi has got shut down, do not disable queues which are,
-		 created based on DNS-SD broadcasts as the server has most
-		 probably not gone away */
-#endif /* HAVE_AVAHI */
-	      disable_printer(p->queue_name,
-			      "Printer disappeared or cups-browsed shutdown");
 	    /* Schedule the removal of the queue for later */
 	    if (in_shutdown == 0) {
               current_time = time(NULL);
