From 2c4fcacd8ba6076f985f88b4587f123f10a8ceb5 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Tuomas=20R=C3=A4s=C3=A4nen?= <tuomas.rasanen@opinsys.fi>
Date: Fri, 26 Apr 2024 11:01:01 +0300
Subject: [PATCH 1/8] [puavo] Disable user-accounts

---
 shell/cc-panel-loader.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/shell/cc-panel-loader.c b/shell/cc-panel-loader.c
index 17f96017f..98b7e93e3 100644
--- a/shell/cc-panel-loader.c
+++ b/shell/cc-panel-loader.c
@@ -136,7 +136,6 @@ static CcPanelLoaderVtable default_panels[] =
 #endif
   PANEL_TYPE("universal-access", cc_ua_panel_get_type,                   NULL),
   PANEL_TYPE("usage",            cc_usage_panel_get_type,                NULL),
-  PANEL_TYPE("user-accounts",    cc_user_panel_get_type,                 NULL),
 #ifdef BUILD_WACOM
   PANEL_TYPE("wacom",            cc_wacom_panel_get_type,                cc_wacom_panel_static_init_func),
 #endif
-- 
2.39.2

