diff -ruN gnome-control-center.orig/shell/cc-panel-loader.c gnome-control-center/shell/cc-panel-loader.c
--- gnome-control-center.orig/shell/cc-panel-loader.c	2022-09-06 13:30:27.566056307 +0300
+++ gnome-control-center/shell/cc-panel-loader.c	2022-09-06 13:32:11.689094656 +0300
@@ -127,7 +127,6 @@
 #endif
   PANEL_TYPE("universal-access", cc_ua_panel_get_type,                   NULL),
   PANEL_TYPE("usage",            cc_usage_panel_get_type,                NULL),
-  PANEL_TYPE("user-accounts",    cc_user_panel_get_type,                 NULL),
 #ifdef BUILD_WACOM
   PANEL_TYPE("wacom",            cc_wacom_panel_get_type,                cc_wacom_panel_static_init_func),
 #endif
