diff -ruN nbd-3.21.orig/nbd-server.c nbd-3.21/nbd-server.c
--- nbd-3.21.orig/nbd-server.c	2023-02-07 15:20:23.000000000 +0200
+++ nbd-3.21/nbd-server.c	2023-02-07 15:25:19.087422186 +0200
@@ -757,6 +757,14 @@
 }
 
 /**
+ * To be called by GArray clearing function.
+ * @param server pointer to server element
+ */
+static void serve_clear_element(SERVER **server) {
+	serve_dec_ref(*server);
+}
+
+/**
  * Parse the config file.
  *
  * @param f the name of the config file
@@ -858,7 +866,7 @@
 	cfile = g_key_file_new();
 	retval = g_array_new(FALSE, TRUE, sizeof(SERVER*));
 	if(expect_generic) {
-		g_array_set_clear_func(retval, (GDestroyNotify)serve_dec_ref);
+		g_array_set_clear_func(retval, (GDestroyNotify)serve_clear_element);
 	}
 	if(!g_key_file_load_from_file(cfile, f, G_KEY_FILE_KEEP_COMMENTS |
 			G_KEY_FILE_KEEP_TRANSLATIONS, &err)) {
@@ -3056,6 +3064,7 @@
                 if (new_server->servename
                     && -1 == get_index_by_servename(new_server->servename,
                                                     servers)) {
+			serve_inc_ref(new_server);
 			g_array_append_val(servers, new_server);
                 }
         }
