--- a/src/plugins/platforms/xcb/qxcbatom.cpp        2023-02-17 18:09:51.394695874 +0200
+++ b/src/plugins/platforms/xcb/qxcbatom.cpp        2023-02-17 18:14:09.167390442 +0200
@@ -90,6 +90,8 @@
 
     "_QT_CLOSE_CONNECTION\0"
 
+    "_QT_GET_TIMESTAMP\0"
+
     "_MOTIF_WM_HINTS\0"
 
     "DTWM_IS_RUNNING\0"
--- a/src/plugins/platforms/xcb/qxcbatom.h  2023-02-17 18:09:51.394695874 +0200
+++ b/src/plugins/platforms/xcb/qxcbatom.h  2023-02-17 18:55:04.234697485 +0200
@@ -91,6 +91,8 @@
         // Qt/XCB specific
         _QT_CLOSE_CONNECTION,
 
+        _QT_GET_TIMESTAMP,
+
         _MOTIF_WM_HINTS,
 
         DTWM_IS_RUNNING,
--- a/src/plugins/platforms/xcb/qxcbclipboard.cpp   2023-02-17 18:09:51.394695874 +0200
+++ b/src/plugins/platforms/xcb/qxcbclipboard.cpp   2023-02-17 18:17:04.167886705 +0200
@@ -829,6 +829,8 @@
         alloc_error = buf.size() != nbytes+1;
     }
 
+    QElapsedTimer timer;
+    timer.start();
     for (;;) {
         connection()->flush();
         xcb_generic_event_t *ge = waitForClipboardEvent(win, XCB_PROPERTY_NOTIFY);
@@ -864,9 +866,11 @@
                 tmp_buf.resize(0);
                 offset += length;
             }
-        } else {
-            break;
         }
+
+        const auto elapsed = timer.elapsed();
+        if (elapsed > clipboard_timeout)
+            break;
     }
 
     // timed out ... create a new requestor window, otherwise the requestor
--- a/src/plugins/platforms/xcb/qxcbconnection.cpp  2023-02-17 18:09:51.394695874 +0200
+++ b/src/plugins/platforms/xcb/qxcbconnection.cpp  2023-02-17 18:15:27.731614360 +0200
@@ -755,8 +755,8 @@
 {
     // send a dummy event to myself to get the timestamp from X server.
     xcb_window_t window = rootWindow();
-    xcb_atom_t dummyAtom = atom(QXcbAtom::CLIP_TEMPORARY);
-    xcb_change_property(xcb_connection(), XCB_PROP_MODE_APPEND, window, dummyAtom,
+    xcb_atom_t dummyAtom = atom(QXcbAtom::_QT_GET_TIMESTAMP);
+    xcb_change_property(xcb_connection(), XCB_PROP_MODE_REPLACE, window, dummyAtom,
                         XCB_ATOM_INTEGER, 32, 0, nullptr);
 
     connection()->flush();
@@ -777,8 +777,6 @@
     xcb_timestamp_t timestamp = pn->time;
     free(event);
 
-    xcb_delete_property(xcb_connection(), window, dummyAtom);
-
     return timestamp;
 }
 

