diff -ruN cups-filters-1.28.17.puavobase.0/utils/cups-browsed.c cups-filters-1.28.17/utils/cups-browsed.c
--- cups-filters-1.28.17.puavobase.0/utils/cups-browsed.c	2024-08-26 13:22:57.282921310 +0300
+++ cups-filters-1.28.17/utils/cups-browsed.c	2024-08-27 08:39:37.857374927 +0300
@@ -8138,28 +8138,10 @@
 	      break;
 	    }
 	  }
-	  /* Error on modifying printer-is-shared bit, removing possibly
+	  /* Error on modifying printer-is-shared bit, remove the
 	     temporary queue */
 	  if (i <= 1) {
 	    debug_printf("Removing the possibly temporary CUPS queue.\n");
-	    /* Check whether there are still jobs and do not remove the queue
-	       then */
-	    num_jobs = 0;
-	    jobs = NULL;
-	    num_jobs = cupsGetJobs2(http, &jobs, p->queue_name, 0,
-				    CUPS_WHICHJOBS_ACTIVE);
-	    if (num_jobs > 0) { /* there are still jobs */
-	      debug_printf("Temporary queue has still jobs or CUPS error, retrying later.\n");
-	      cupsFreeJobs(num_jobs, jobs);
-	      /* Schedule the removal of the queue for later */
-	      if (in_shutdown == 0) {
-                current_time = time(NULL);
-		p->timeout = current_time + TIMEOUT_RETRY;
-		p->no_autosave = 0;
-	      }
-	      break;
-	    }
-	    /* No jobs, remove the CUPS queue */
 	    request = ippNewRequest(CUPS_DELETE_PRINTER);
 	    ippAddString(request, IPP_TAG_OPERATION, IPP_TAG_URI,
 			 "printer-uri", NULL, uri);
