description: don't append wine to the default path settings
author: Michael Gilbert <mgilbert@debian.org>

--- a/Makefile.in
+++ b/Makefile.in
@@ -28,8 +28,8 @@ datarootdir     = @datarootdir@
 datadir         = @datadir@
 mandir          = @mandir@
 includedir      = @includedir@
-fontdir         = ${datadir}/wine/fonts
-dlldir          = ${libdir}/wine
+fontdir         = /usr/share/wine/fonts
+dlldir          = ${libdir}
 top_srcdir      = @top_srcdir@
 top_builddir    = @top_builddir@
 srcdir          = @srcdir@
