diff -ruN mutter/src/backends/meta-monitor-manager.c mutter.2/src/backends/meta-monitor-manager.c
--- mutter/src/backends/meta-monitor-manager.c	2025-11-22 20:16:29.000000000 +0100
+++ mutter/src/backends/meta-monitor-manager.c	2026-06-15 11:49:16.896337850 +0200
@@ -663,6 +663,7 @@
     case META_MONITORS_CONFIG_METHOD_PERSISTENT:
       meta_monitor_config_manager_set_current (manager->config_manager, config);
       break;
+    case META_MONITORS_CONFIG_METHOD_STOREONLY:
     case META_MONITORS_CONFIG_METHOD_VERIFY:
       break;
     }
@@ -2192,9 +2193,10 @@
 }
 
 static gboolean
-meta_monitor_manager_handle_get_current_state (MetaDBusDisplayConfig *skeleton,
-                                               GDBusMethodInvocation *invocation,
-                                               MetaMonitorManager    *manager)
+meta_monitor_manager_handle_current_state (MetaDBusDisplayConfig *skeleton,
+                                           GDBusMethodInvocation *invocation,
+                                           MetaMonitorManager    *manager,
+                                           gboolean return_dbus_answer)
 {
   GVariantBuilder monitors_builder;
   GVariantBuilder logical_monitors_builder;
@@ -2438,17 +2440,28 @@
                              g_variant_builder_end (&max_screen_size_builder));
     }
 
-  meta_dbus_display_config_complete_get_current_state (
-    skeleton,
-    invocation,
-    manager->serial,
-    g_variant_builder_end (&monitors_builder),
-    g_variant_builder_end (&logical_monitors_builder),
-    g_variant_builder_end (&properties_builder));
+
+  if (return_dbus_answer) {
+    meta_dbus_display_config_complete_get_current_state (
+      skeleton,
+      invocation,
+      manager->serial,
+      g_variant_builder_end (&monitors_builder),
+      g_variant_builder_end (&logical_monitors_builder),
+      g_variant_builder_end (&properties_builder));
+  }
 
   return G_DBUS_METHOD_INVOCATION_HANDLED;
 }
 
+static gboolean
+meta_monitor_manager_handle_get_current_state (MetaDBusDisplayConfig *skeleton,
+                                               GDBusMethodInvocation *invocation,
+                                               MetaMonitorManager    *manager)
+{
+  return meta_monitor_manager_handle_current_state (skeleton, invocation, manager, 1);
+}
+
 #undef MODE_FORMAT
 #undef MODES_FORMAT
 #undef MONITOR_SPEC_FORMAT
@@ -3011,6 +3024,18 @@
   GList *for_lease_monitor_specs = NULL;
   GError *error = NULL;
 
+  if (method == META_MONITORS_CONFIG_METHOD_STOREONLY) {
+    if (meta_monitor_manager_handle_current_state (skeleton, invocation, manager, 0)) {
+      meta_monitor_config_manager_save_current (manager->config_manager);
+      g_dbus_method_invocation_return_value (invocation, g_variant_new ("()"));
+    } else {
+      g_dbus_method_invocation_return_error (invocation, G_DBUS_ERROR,
+                                             G_DBUS_ERROR_FAILED,
+                                             "Problem in saving the configuration");
+    }
+    return TRUE;
+  }
+
   if (serial != manager->serial)
     {
       g_dbus_method_invocation_return_error (invocation, G_DBUS_ERROR,
diff -ruN mutter/src/backends/meta-monitor-manager-private.h mutter.2/src/backends/meta-monitor-manager-private.h
--- mutter/src/backends/meta-monitor-manager-private.h	2025-11-22 20:16:29.000000000 +0100
+++ mutter/src/backends/meta-monitor-manager-private.h	2026-06-15 11:46:18.652850267 +0200
@@ -56,7 +56,8 @@
 {
   META_MONITORS_CONFIG_METHOD_VERIFY = 0,
   META_MONITORS_CONFIG_METHOD_TEMPORARY = 1,
-  META_MONITORS_CONFIG_METHOD_PERSISTENT = 2
+  META_MONITORS_CONFIG_METHOD_PERSISTENT = 2,
+  META_MONITORS_CONFIG_METHOD_STOREONLY = 3
 } MetaMonitorsConfigMethod;
 
 /* Equivalent to the 'layout-mode' enum in org.gnome.Mutter.DisplayConfig */
