diff -ruN mutter/src/backends/meta-monitor-manager.c mutter.3/src/backends/meta-monitor-manager.c
--- mutter/src/backends/meta-monitor-manager.c	2026-06-15 11:49:16.896337850 +0200
+++ mutter/src/backends/meta-monitor-manager.c	2026-06-15 11:58:05.194045648 +0200
@@ -781,6 +781,8 @@
   MetaMonitorsConfigMethod method;
   MetaMonitorsConfigMethod fallback_method =
     META_MONITORS_CONFIG_METHOD_TEMPORARY;
+  MetaMonitorSwitchConfigType config_type;
+  char *display_setup;
 
   use_stored_config = should_use_stored_config (manager);
   if (use_stored_config)
@@ -929,7 +931,17 @@
         }
     }
 
-  config = meta_monitor_config_manager_create_linear (manager->config_manager);
+  display_setup = getenv("MUTTER_DISPLAY_SETUP");
+  config_type =
+    display_setup
+      ? (strcmp(display_setup, "all-mirror") == 0 ? META_MONITOR_SWITCH_CONFIG_ALL_MIRROR :
+         strcmp(display_setup, "external"  ) == 0 ? META_MONITOR_SWITCH_CONFIG_EXTERNAL   :
+         strcmp(display_setup, "builtin"   ) == 0 ? META_MONITOR_SWITCH_CONFIG_BUILTIN    :
+                                                    META_MONITOR_SWITCH_CONFIG_ALL_LINEAR)
+      : META_MONITOR_SWITCH_CONFIG_ALL_LINEAR;
+
+  config = meta_monitor_config_manager_create_for_switch_config(
+    manager->config_manager, config_type);
   if (config)
     {
       if (!meta_monitor_manager_apply_monitors_config (manager,
